{"573408419": {"id": 573408419, "post_id": 573408419, "username": "zahin-mohammad", "body": "", "created_at": 1583009138, "updated_at": 1583009138}, "549249843": {"id": 549249843, "post_id": 549249843, "username": "zahin-mohammad", "body": "If the script is run within a folder that is not at `~/` then the wrong path will be assigned from the following lines \r\n```\r\n\techo \"\\e[45m Installing zsh-autosuggestions \\e[0m\"\r\n\t# Install zsh-autosuggestions (not via oh-my-zsh)\r\n\tgit clone https://github.com/zsh-users/zsh-autosuggestions ~/.zsh/zsh-autosuggestions\r\n\techo \"source ${(q-)PWD}/.zsh/zsh-autosuggestions/zsh-autosuggestions.zsh\" >> ${ZDOTDIR:-$HOME}/.zshrc\r\n\r\n\r\n\techo \"\\e[45m Installing zsh-syntax-highlighting \\e[0m\"\r\n\t# Install zsh-syntax-highlighting (not via oh-my-zsh)\r\n\tgit clone https://github.com/zsh-users/zsh-syntax-highlighting.git ~/.zsh/zsh-syntax-highlighting\r\n\techo \"source ${(q-)PWD}/.zsh/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh\" >> ${ZDOTDIR:-$HOME}/.zshrc\r\n```", "created_at": 1578959548, "updated_at": 1578959574}, "545331007": {"id": 545331007, "post_id": 545331007, "username": "zahin-mohammad", "body": "", "created_at": 1578170967, "updated_at": 1578170967}, "510116274": {"id": 510116274, "post_id": 510116274, "username": "dependabot[bot]", "body": "Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.23 to 1.24.2.\n<details>\n<summary>Changelog</summary>\n\n*Sourced from [urllib3's changelog](https://github.com/urllib3/urllib3/blob/master/CHANGES.rst).*\n\n> 1.24.2 (2019-04-17)\n> -------------------\n> \n> * Don't load system certificates by default when any other ``ca_certs``, ``ca_certs_dir`` or\n>   ``ssl_context`` parameters are specified.\n> \n> * Remove Authorization header regardless of case when redirecting to cross-site. (Issue [#1510](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1510))\n> \n> * Add support for IPv6 addresses in subjectAltName section of certificates. (Issue [#1269](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1269))\n> \n> \n> 1.24.1 (2018-11-02)\n> -------------------\n> \n> * Remove quadratic behavior within ``GzipDecoder.decompress()`` (Issue [#1467](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1467))\n> \n> * Restored functionality of ``ciphers`` parameter for ``create_urllib3_context()``. (Issue [#1462](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1462))\n> \n> \n> 1.24 (2018-10-16)\n> -----------------\n> \n> * Allow key_server_hostname to be specified when initializing a PoolManager to allow custom SNI to be overridden. (Pull [#1449](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1449))\n> \n> * Test against Python 3.7 on AppVeyor. (Pull [#1453](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1453))\n> \n> * Early-out ipv6 checks when running on App Engine. (Pull [#1450](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1450))\n> \n> * Change ambiguous description of backoff_factor (Pull [#1436](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1436))\n> \n> * Add ability to handle multiple Content-Encodings (Issue [#1441](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1441) and Pull [#1442](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1442))\n> \n> * Skip DNS names that can't be idna-decoded when using pyOpenSSL (Issue [#1405](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1405)).\n> \n> * Add a server_hostname parameter to HTTPSConnection which allows for\n>   overriding the SNI hostname sent in the handshake. (Pull [#1397](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1397))\n> \n> * Drop support for EOL Python 2.6 (Pull [#1429](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1429) and Pull [#1430](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1430))\n> \n> * Fixed bug where responses with header Content-Type: message/* erroneously\n>   raised HeaderParsingError, resulting in a warning being logged. (Pull [#1439](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1439))\n> \n> * Move urllib3 to src/urllib3 (Pull [#1409](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1409))\n</details>\n<details>\n<summary>Commits</summary>\n\n- [`1efadf4`](https://github.com/urllib3/urllib3/commit/1efadf43dc63317cd9eaa3e0fdb9e05ab07254b1) Release 1.24.2 ([#1564](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1564))\n- [`a6ec68a`](https://github.com/urllib3/urllib3/commit/a6ec68a5c5c5743c59fe5c62c635c929586c429b) Merging new release version: 1.24.1\n- [`0cedb3b`](https://github.com/urllib3/urllib3/commit/0cedb3b0f1e5d79c89c6db767c534b064b794cf2) Restore context.set_ciphers() to create_urllib3_context() ([#1463](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1463))\n- [`0aeba3b`](https://github.com/urllib3/urllib3/commit/0aeba3be0224a930f6ffef254ed12b41303a86d7) Use bytearray to accumulate bytes from gzip ([#1468](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1468))\n- [`f8d1c78`](https://github.com/urllib3/urllib3/commit/f8d1c787d9b02a70d66ddbde9c99061d9073d54a) Uninstall oclint to ensure gcc can be brew upgraded ([#1464](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1464))\n- [`cd7cfa6`](https://github.com/urllib3/urllib3/commit/cd7cfa613b2678e700597d098ce9bbdc934863e6) Resolve pytest pluggy version conflict ([#1457](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1457))\n- [`b548abc`](https://github.com/urllib3/urllib3/commit/b548abc9812a628d3290d0cab83e44f3c31ac6fe) Update changelog for 1.24 release\n- [`ef0c745`](https://github.com/urllib3/urllib3/commit/ef0c74542abe69421a86c4d3c6a86fe43cb809a4) Merging new release version: 1.24\n- [`a0964d9`](https://github.com/urllib3/urllib3/commit/a0964d9947c07d2b8495726ac23ad251b5d236af) Add missing key_server_hostname variable ([#1449](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1449))\n- [`34d8298`](https://github.com/urllib3/urllib3/commit/34d8298ecf93c84f9916457d89701d6d5c807780) Test against Python 3.7 on AppVeyor ([#1453](https://github-redirect.dependabot.com/urllib3/urllib3/issues/1453))\n- Additional commits viewable in [compare view](https://github.com/urllib3/urllib3/compare/1.23...1.24.2)\n</details>\n<br />\n\n[![Dependabot compatibility score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=urllib3&package-manager=pip&previous-version=1.23&new-version=1.24.2)](https://help.github.com/articles/configuring-automated-security-fixes)\n\nDependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting `@dependabot rebase`.\n\n[//]: # (dependabot-automerge-start)\n[//]: # (dependabot-automerge-end)\n\n---\n\n<details>\n<summary>Dependabot commands and options</summary>\n<br />\n\nYou can trigger Dependabot actions by commenting on this PR:\n- `@dependabot rebase` will rebase this PR\n- `@dependabot recreate` will recreate this PR, overwriting any edits that have been made to it\n- `@dependabot merge` will merge this PR after your CI passes on it\n- `@dependabot squash and merge` will squash and merge this PR after your CI passes on it\n- `@dependabot cancel merge` will cancel a previously requested merge and block automerging\n- `@dependabot reopen` will reopen this PR if it is closed\n- `@dependabot ignore this [patch|minor|major] version` will close this PR and stop Dependabot creating any more for this minor/major version (unless you reopen the PR or upgrade to it yourself)\n- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)\n- `@dependabot use these labels` will set the current labels as the default for future PRs for this repo and language\n- `@dependabot use these reviewers` will set the current reviewers as the default for future PRs for this repo and language\n- `@dependabot use these assignees` will set the current assignees as the default for future PRs for this repo and language\n- `@dependabot use this milestone` will set the current milestone as the default for future PRs for this repo and language\n\nYou can disable automated security fix PRs for this repo from the [Security Alerts page](https://github.com/atulbipin/spotify-slackbot/network/alerts).\n\n</details>", "created_at": 1571676788, "updated_at": 1571676789}, "464977589": {"id": 464977589, "post_id": 464977589, "username": "zahin-mohammad", "body": "", "created_at": 1562522669, "updated_at": 1562522669}, "464977554": {"id": 464977554, "post_id": 464977554, "username": "zahin-mohammad", "body": "", "created_at": 1562522649, "updated_at": 1562522649}, "11843860": {"id": 11843860, "post_id": 11843860, "username": "AttwellBrian", "body": "Clean this up some time. And follow this: Read http://opencv.org/platforms/android/android-best-practices.html before making modifications.\n", "created_at": 1362861825, "updated_at": 1362861827}, "581340665": {"id": 581340665, "post_id": 581340665, "username": "dependabot[bot]", "body": "Bumps [acorn](https://github.com/acornjs/acorn) from 5.7.3 to 5.7.4.\n<details>\n<summary>Commits</summary>\n<ul>\n<li><a href=\"https://github.com/acornjs/acorn/commit/6370e90067552022710190319cbbbd8c43001957\"><code>6370e90</code></a> Mark version 5.7.4</li>\n<li><a href=\"https://github.com/acornjs/acorn/commit/fbc15b1344f6dfb992f67b4bbf1357436247c8a0\"><code>fbc15b1</code></a> More rigorously check surrogate pairs in regexp validator</li>\n<li>See full diff in <a href=\"https://github.com/acornjs/acorn/compare/5.7.3...5.7.4\">compare view</a></li>\n</ul>\n</details>\n<br />\n\n\n[![Dependabot compatibility score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=acorn&package-manager=npm_and_yarn&previous-version=5.7.3&new-version=5.7.4)](https://help.github.com/articles/configuring-automated-security-fixes)\n\nDependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting `@dependabot rebase`.\n\n[//]: # (dependabot-automerge-start)\n[//]: # (dependabot-automerge-end)\n\n---\n\n<details>\n<summary>Dependabot commands and options</summary>\n<br />\n\nYou can trigger Dependabot actions by commenting on this PR:\n- `@dependabot rebase` will rebase this PR\n- `@dependabot recreate` will recreate this PR, overwriting any edits that have been made to it\n- `@dependabot merge` will merge this PR after your CI passes on it\n- `@dependabot squash and merge` will squash and merge this PR after your CI passes on it\n- `@dependabot cancel merge` will cancel a previously requested merge and block automerging\n- `@dependabot reopen` will reopen this PR if it is closed\n- `@dependabot close` will close this PR and stop Dependabot recreating it. You can achieve the same result by closing it manually\n- `@dependabot ignore this major version` will close this PR and stop Dependabot creating any more for this major version (unless you reopen the PR or upgrade to it yourself)\n- `@dependabot ignore this minor version` will close this PR and stop Dependabot creating any more for this minor version (unless you reopen the PR or upgrade to it yourself)\n- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)\n- `@dependabot use these labels` will set the current labels as the default for future PRs for this repo and language\n- `@dependabot use these reviewers` will set the current reviewers as the default for future PRs for this repo and language\n- `@dependabot use these assignees` will set the current assignees as the default for future PRs for this repo and language\n- `@dependabot use this milestone` will set the current milestone as the default for future PRs for this repo and language\n\nYou can disable automated security fix PRs for this repo from the [Security Alerts page](https://github.com/AttwellBrian/nico/network/alerts).\n\n</details>", "created_at": 1584214355, "updated_at": 1584214356}, "383727930": {"id": 383727930, "post_id": 383727930, "username": "li-boxuan", "body": "Hi, this dataset seems interesting, but the latest update is 3 years ago.\r\n\r\nJust wondering if there is any plan on updates.", "created_at": 1542959386, "updated_at": 1542959386}, "517043197": {"id": 517043197, "post_id": 517043197, "username": "anil-goudar", "body": "How to specify the vertical limit so that the column contents can be modified accordingly.", "created_at": 1572860880, "updated_at": 1572860880}, "481923312": {"id": 481923312, "post_id": 481923312, "username": "tanyachawla412", "body": "", "created_at": 1566073628, "updated_at": 1566073628}, "478103608": {"id": 478103608, "post_id": 478103608, "username": "margarineHound", "body": "I want the bullet points for each position to line up with the position title and employer. I have modified the \\tightitemize command as follows: \r\n\r\n`\\newenvironment{tightitemize} % Defines the tightitemize environment which modifies the itemize environment to be more compact\r\n{\\vspace{-\\topsep}\\begin{itemize} \\leftmargin-25pt \\itemsep0pt \\parskip0pt \\parsep0pt \\itemindent-20pt \\labelsep1pt \\parindent0pt\r\n}\r\n{\\end{itemize}\\vspace{-\\topsep}}`\r\n\r\nHowever, it seems that this is causing a hanging indent: ie. I want the second line of each bullet point to line up vertically with the first word in the first line, if not the bullet. \r\n\r\n![Rishabh CV_v3 - Online LaTeX Editor Overleaf - Mozilla Firefox 8_7_2019 01_02_38_LI_Moment](https://user-images.githubusercontent.com/7854019/62652271-00385400-b929-11e9-961a-42df41e9a32a.jpg)\r\n\r\nI have tried playing around with the values, and parameters, to no avail. I would appreciate any and all help. \r\nThank you in advance. \r\n\r\n", "created_at": 1565206562, "updated_at": 1565206562}, "448359411": {"id": 448359411, "post_id": 448359411, "username": "hengdashi", "body": "", "created_at": 1558733291, "updated_at": 1564500122}, "516467450": {"id": 516467450, "post_id": 448359411, "username": "AngelOnFira", "body": "Thanks for this :100:", "created_at": 1564500122, "updated_at": 1564500122}, "412694458": {"id": 412694458, "post_id": 412694458, "username": "voidstarstar", "body": "Fixes the overlap between the section name and body.\r\nFixes #42", "created_at": 1550709335, "updated_at": 1550709335}, "409262011": {"id": 409262011, "post_id": 409262011, "username": "utkuturk", "body": "Hello, thank you for your work. It is awesome! Is there a way to write \u010d this character? I could not did it. When I try writing `\\v{c}` it gives a question mark in the place of the character.\r\n\r\nThank you for your work.", "created_at": 1549972576, "updated_at": 1555174481}, "482838797": {"id": 482838797, "post_id": 409262011, "username": "spilehchiha", "body": "For an initial ad-hoc solution, you can try including this command to your preamble:\r\n`\\usepackage[T1]{fontenc}`\r\nUnfortunately, it changes the default font, but it gets the job done and recognizes the upper `v` above the `c` character.", "created_at": 1555174481, "updated_at": 1555174481}, "350898657": {"id": 350898657, "post_id": 350898657, "username": "yadavabhinav16", "body": "I tried using this template in both overleaf and shareLaTeX. \r\n\\textbf{ sample text } doesn't seem to work. Anything that I might be missing on?", "created_at": 1534354650, "updated_at": 1540838043}, "434028919": {"id": 434028919, "post_id": 350898657, "username": "peey", "body": "See https://tex.stackexchange.com/a/274901/144456\r\n\r\nThe solution is to use `\\custombold{ ... }` instead.\r\n\r\n", "created_at": 1540838043, "updated_at": 1540838043}, "318642291": {"id": 318642291, "post_id": 318642291, "username": "prateekiiest", "body": "Is there any way of modification of colors ?\r\nSince in some areas it kind of gets faded out", "created_at": 1524926755, "updated_at": 1527195519}, "391859522": {"id": 391859522, "post_id": 318642291, "username": "tonystonee", "body": "I agree with this question, I try to change the color of the \\section and get error:\r\n`Package xcolor Error: Undefined color `RED'.`\r\n\r\nI can't seem to find a work around, anyone know of one?\r\n", "created_at": 1527195519, "updated_at": 1527195519}, "307608049": {"id": 307608049, "post_id": 307608049, "username": "DeeprajPandey", "body": "Added a height argument to \\begin{minipage}. When the contents are more than the vertical limit, the extra contents will run into the page instead of overflowing to the next page.\r\nThis can be a solution till we figure out a way to smoothly transition to the next page.\r\n\r\nAlso added a % after the first \\end{minipage} which removes the little bit of horizontal space that is automatically inserted by the `minipage` environment. [Reference](https://tex.stackexchange.com/questions/157222/putting-figures-side-by-side-using-minipage)", "created_at": 1521718822, "updated_at": 1521718989}, "298012184": {"id": 298012184, "post_id": 298012184, "username": "srdg", "body": "I used this template and while making my CV, I felt it would be a cleaner design if we could separate the two columns with a single `vline`, with width, say, `0.4 pts`. Can that be included anywhere?", "created_at": 1518881029, "updated_at": 1518881029}, "291973577": {"id": 291973577, "post_id": 291973577, "username": "niranjantdesai", "body": "I'm using the OpenFonts template on Windows with TeXworks for compilation. There's a problem with the \"Publications\" section --- the body and the section name are overlapping.\r\n\r\n![capture](https://user-images.githubusercontent.com/15803433/35452041-45161af8-02eb-11e8-92bb-a6b6f105370d.PNG)\r\n\r\nA better fix seems to be using this:\r\n`\\renewcommand\\refname{PUBLICATIONS}`\r\ninstead of the current code, which is\r\n`\\section{Publications}`\r\n`\\renewcommand\\refname{\\vskip -1.5cm} % Couldn't get this working from the .cls file`\r\n\r\n![capture1](https://user-images.githubusercontent.com/15803433/35452188-bcd015c6-02eb-11e8-8d83-c1517ba7d5fd.PNG)", "created_at": 1516987547, "updated_at": 1550709802}, "465815005": {"id": 465815005, "post_id": 291973577, "username": "voidstarstar", "body": "I didn't see any pull request for this issue, so I created #50.  Thanks for providing the solution.", "created_at": 1550709802, "updated_at": 1550709802}, "277975364": {"id": 277975364, "post_id": 277975364, "username": "anthonyattard", "body": "If you believe this belongs in a separate repo let me know.", "created_at": 1512008341, "updated_at": 1512008341}, "273544317": {"id": 273544317, "post_id": 273544317, "username": "naveenT1010", "body": "Hi,\r\nHow can I enable the format to extend into two pages, without first one being blank?\r\nThanks", "created_at": 1510600425, "updated_at": 1522068507}, "376153699": {"id": 376153699, "post_id": 273544317, "username": "rohit-upadhya", "body": "You can follow what I have done with my CV. I have taken it from this repo only and edited it. \r\nhttps://github.com/nocturnalnocturnal/ROHIT_UPADHYA_CV", "created_at": 1522068507, "updated_at": 1522068507}, "265499138": {"id": 265499138, "post_id": 265499138, "username": "csaund", "body": "Right now without any special formatting, any overflow onto a second page is poorly formatted. A version of this which allows experience to take up the full column on the first page and awards, leadership, etc to fully occupy the second would keep the clean style and continue to emphasize work experience. \r\n\r\nHave you thought about adapting this to look good on 2 pages? ", "created_at": 1507994894, "updated_at": 1545691926}, "339554756": {"id": 339554756, "post_id": 265499138, "username": "juliocesar-io", "body": "Hi @csaund, you can create any pages you want like this (just keep the same `textwidth`  for the correspondent column)\r\n```\r\n\\begin{document}\r\n\r\n% PAGE ONE\r\n\r\n\\begin{minipage}[t]{0.38\\textwidth}\r\n% COLUMN ONE SECTIONS\r\n\\end{minipage}\r\n\r\n\\begin{minipage}[t]{0.61\\textwidth}\r\n% COLUMN TWO SECTIONS\r\n\\end{minipage}\r\n\r\n\\newpage\r\n\r\n% PAGE TWO\r\n\r\n\\begin{minipage}[t]{0.38\\textwidth}\r\n% COLUMN ONE SECTIONS\r\n\\end{minipage}\r\n\r\n\\begin{minipage}[t]{0.61\\textwidth}\r\n% COLUMN TWO SECTIONS\r\n\\end{minipage}\r\n\r\n% PAGE N ...\r\n\r\n\\end{document}\r\n```", "created_at": 1508995406, "updated_at": 1509310942}, "404833506": {"id": 404833506, "post_id": 265499138, "username": "arp1561", "body": "The above-mentioned way and adding \\hfill between mini pages makes the deedy resume work with 2 pages perfectly", "created_at": 1531488523, "updated_at": 1531488523}, "449776182": {"id": 449776182, "post_id": 265499138, "username": "trenta3", "body": "I would also like to suggest the paracol package with which the splitting is much more easy to do without having to adjust the minipages each time you modify the CV. The document would look something like:\r\n```\r\n\\begin{document}\r\n\r\n\\columnratio{0.44}\r\n\\begin{paracol}{2}\r\n% LATERAL CONTENT\r\n\\switchcolumn\r\n% CENTRAL CONTENT\r\n\\end{paracol}\r\n```", "created_at": 1545691926, "updated_at": 1545691926}, "261485044": {"id": 261485044, "post_id": 261485044, "username": "anthonyattard", "body": "Does anyone know of any templates that are stylistically similar to Deedy? ", "created_at": 1506639220, "updated_at": 1518661793}, "343696950": {"id": 343696950, "post_id": 261485044, "username": "wilfredom", "body": "Yes-ish: [AwesomeCV Cover letter](https://www.sharelatex.com/templates/cover-letters/awesome-cv-cover-letter).\r\n\r\nAll instances of red can be changed to black near the beginning of the document: look for the line `\\colorlet{awesome}{awesome-red}` and change `red` to `darknight`.\r\n\r\nI also tried getting the name displayed in the same font by copying over deedy fonts and changing them in the .cls, but for some reason the last name comes out much bolder. ", "created_at": 1510437538, "updated_at": 1510437538}, "346216956": {"id": 346216956, "post_id": 261485044, "username": "anthonyattard", "body": "Thanks for the response.", "created_at": 1511314834, "updated_at": 1511314834}, "347975932": {"id": 347975932, "post_id": 261485044, "username": "wilfredom", "body": "No problem! Another strategy (which I'm testing out right now) is to adapt the [single-column version](https://www.overleaf.com/latex/templates/single-column-deedy-cv-slash-resume-template/zwyxmkbrfgtz) of Deedy, by Overleaf.", "created_at": 1511985058, "updated_at": 1511985058}, "348061332": {"id": 348061332, "post_id": 261485044, "username": "anthonyattard", "body": "Checkout my cover letter template https://github.com/anthonyattard/Deedy-Resume", "created_at": 1512008220, "updated_at": 1512008220}, "348062553": {"id": 348062553, "post_id": 261485044, "username": "wilfredom", "body": "Wow, fantastic. Thanks so much, Anthony!", "created_at": 1512008676, "updated_at": 1512008676}, "365809119": {"id": 365809119, "post_id": 261485044, "username": "anthonyattard", "body": "Any chance this can be merged?", "created_at": 1518661793, "updated_at": 1518661793}, "254465607": {"id": 254465607, "post_id": 254465607, "username": "wrmacrae", "body": "See other PR.", "created_at": 1504212998, "updated_at": 1504212998}, "254465081": {"id": 254465081, "post_id": 254465081, "username": "wrmacrae", "body": "I don't think anyone uses \"Github://\" outside of these resumes. If I'm wrong and this is a real thing, feel free to decline. But it seems weird.", "created_at": 1504212879, "updated_at": 1504212879}, "208788231": {"id": 208788231, "post_id": 208788231, "username": "joshjacobson", "body": "When column 2 and column 1 are the same length (or so they appear, perhaps there is a very small length difference causing this), column 2 causes overflow to the next page, while column 1 does not. \r\n\r\nBasically, I am able to have 1 additional line on column 1 than I am on column 2. I would like to adjust it so this last line is allowed in column 2 as well.", "created_at": 1487569123, "updated_at": 1487569123}, "190428237": {"id": 190428237, "post_id": 190428237, "username": "kfeine", "body": "Hi,\r\n\r\nIt seems there is an issue with small caps in `\\descript{}`: some letters with accents are turned to lowercase instead.\r\n\r\n[Here is an example](http://hpics.li/5672503) with:\r\n\r\n    \\descript{Ae\u00e9e\u00e8u\u00f9a\u00e0}\r\n\r\nI don't think it's directly related with Relaway font, since it supports small caps with accents. Maybe it's more an [issue with selectfont](https://tex.stackexchange.com/questions/286047/tex4ht-getting-accented-letters-in-small-capitals-through-selectfont)?\r\n\r\nThanks,\r\nKind regards,\r\n", "created_at": 1479505223, "updated_at": 1508523798}, "333130640": {"id": 333130640, "post_id": 190428237, "username": "reyhanhgh", "body": "Hi, \r\nI have the same problem. Does anybody know how to fix it?\r\n\r\nhere is the source code: \r\nhttps://www.overleaf.com/latex/templates/deedy-resume/sqdbztjjghvz#.Wc4HfBOCzVp\r\n\r\nMany thanks!", "created_at": 1506692796, "updated_at": 1506692893}, "338285717": {"id": 338285717, "post_id": 190428237, "username": "remidebette", "body": "Hi everyone,\r\n\r\nThe problem seems indeed to be due to the Raleway font.\r\nI opened a ticket there a while back detailing the issue: [Support for accented small caps](https://github.com/impallari/Raleway/issues/21)\r\n\r\nR\u00e9mi", "created_at": 1508523798, "updated_at": 1508523798}, "183199388": {"id": 183199388, "post_id": 183199388, "username": "Data-drone", "body": "I have tried putting the files in:\n`~/.fonts/`\n`/usr/local/share/fonts`\n'/usr/share/fonts/truetype'\n\nI have refreshed the `fc-cache`.\n\nand `fc-list|grep` returns the files\n\nbut fontspec and kpathea can't find them?\n", "created_at": 1476523529, "updated_at": 1476523529}, "175121549": {"id": 175121549, "post_id": 175121549, "username": "jck", "body": "", "created_at": 1473105017, "updated_at": 1473143416}, "244861096": {"id": 244861096, "post_id": 175121549, "username": "nirmitgoyal", "body": "@deedy : nice changes. Deedy please merge this pull request\n", "created_at": 1473143416, "updated_at": 1473143416}, "145788671": {"id": 145788671, "post_id": 145788671, "username": "alanyee", "body": "Under \"Availability\" in the README.md, the Overleaf link links to a source template that is Version 1.1 (30/4/2014). I think this link should be updated to Version 1.2.\n", "created_at": 1459800099, "updated_at": 1459800099}, "142323881": {"id": 142323881, "post_id": 142323881, "username": "kirkang", "body": "1)Add xeCJK, it is supported to type Chinese character. Korean and Japanese are also supported theoretical, but it has not been tested.\n2)Add new command \"the position applied\"\n", "created_at": 1458560666, "updated_at": 1458560666}, "127175342": {"id": 127175342, "post_id": 127175342, "username": "isometric", "body": "I did a spin on your design to simplify the topography and alignment. \n\nSee if you like it :)\n\nhttps://drive.google.com/file/d/0B1mMiFxu14yFdzJIaGtUYlJhUm8/view?usp=docslist_api\n\nSadly I'm not familiar enough with LaTeX to so it was made it in InDesign but you can find the project file here: \n\nhttps://drive.google.com/file/d/0B1mMiFxu14yFV0paNVdIaHliaVU/view?usp=docslist_api\n", "created_at": 1453105203, "updated_at": 1479081239}, "172885081": {"id": 172885081, "post_id": 127175342, "username": "deedy", "body": "I quite like it. I'm no designer, but I think the lowercase headings for each section look nice, although the bold date ranges and location lines seem a little overbearing. The slightly bigger name section looks a bit cleaner but the margins seem a little large around it. I also think the descriptions would look better with bullets so the reader doesnt get lost in a wall of text (even though its paragraphed). Cool stuff though!\n", "created_at": 1453216743, "updated_at": 1453216743}, "260222929": {"id": 260222929, "post_id": 127175342, "username": "pratnala", "body": "@isometric Your files don't exist anymore?\n", "created_at": 1479081239, "updated_at": 1479081239}, "123607701": {"id": 123607701, "post_id": 123607701, "username": "nickyr", "body": "This file would go in the OpenFonts directory also I guess.\n", "created_at": 1450851263, "updated_at": 1450851470}, "123607114": {"id": 123607114, "post_id": 123607114, "username": "nickyr", "body": "Right now, the alignment of employers and positions is a little off. There's an extra space before the \"|\".\n\nBy changing\n\n``` tex\n\\newcommand{\\runsubsection}[1]{\\color{subheadings}\n\\fontspec{Helvetica}\\fontsize{12pt}{12pt}\\selectfont\\bfseries\\uppercase {#1} \\normalfont}\n```\n\nto\n\n``` tex\n\\newcommand{\\runsubsection}[1]{\\color{subheadings}\n\\fontspec{Helvetica}\\fontsize{12pt}{12pt}\\selectfont\\bfseries\\uppercase {#1}\\normalfont}\n```\n\n(removing the space after #1), the extra space goes away.\n\nBefore, with underscores for spaces: \n`EMPLOYER__|_Position`\nAfter: \n`EMPLOYER_|_Position`\n", "created_at": 1450850711, "updated_at": 1487568984}, "280992272": {"id": 280992272, "post_id": 123607114, "username": "joshjacobson", "body": "I think I actually prefer the two spaces. Perhaps this was an intentional design decision?", "created_at": 1487568984, "updated_at": 1487568984}, "92031907": {"id": 92031907, "post_id": 92031907, "username": "SquaredPi", "body": "I get the following errors:\n- Undefined control sequence. [\\sectionspace]\n- LaTeX Error: Environment tightitemize undefined. [\\begin{tightitemize}]\n\nAlso, when I try to redefine the tightitemize environment into the document, I get an \"already defined environment\".\n\nPlease advice.\nThank you\n", "created_at": 1435655287, "updated_at": 1495940774}, "131224023": {"id": 131224023, "post_id": 92031907, "username": "rhuddles", "body": "Having the same issue\n", "created_at": 1439582798, "updated_at": 1439582798}, "143951244": {"id": 143951244, "post_id": 92031907, "username": "CorbanR", "body": "same issue\n", "created_at": 1443505429, "updated_at": 1443505429}, "276460102": {"id": 276460102, "post_id": 92031907, "username": "hawkins", "body": "Have any of you tried `\\sectionsep`?", "created_at": 1485889744, "updated_at": 1485889744}, "304489671": {"id": 304489671, "post_id": 92031907, "username": "MarcoGiancarli", "body": "Replace all instances of `\\sectionspace` with `\\sectionsep` and all instances of `\\tightitemize` with `\\tightemize`", "created_at": 1495940774, "updated_at": 1495940774}, "62188066": {"id": 62188066, "post_id": 62188066, "username": "rrshaban", "body": "\\emph and \\textit do nothing within any part of the template, and I haven't been able to figure out why. \n", "created_at": 1426534802, "updated_at": 1447765627}, "157365714": {"id": 157365714, "post_id": 62188066, "username": "JanetMatsen", "body": "Adding \\setmainfont{Lato Light} to the preamble made \\textit{} work in the Overleaf Version.  https://www.overleaf.com/latex/templates/deedy-resume/sqdbztjjghvz#.VksmZmQrJcw \n", "created_at": 1447765616, "updated_at": 1447765616}, "57681975": {"id": 57681975, "post_id": 57681975, "username": "Void2258", "body": "I did my undergrad at the Rochester Institute of Technology. The template puts the name on 2 lines, which is fine, but does so with atrocious spacing, as it spaces out the first three words to occupy the same space as University of Pittsburgh (my grad school). The end result looks horrible, no matter what I do.\n", "created_at": 1423891489, "updated_at": 1426913614}, "84255137": {"id": 84255137, "post_id": 57681975, "username": "Travis-S", "body": "I had the same problem with my institution, and ended up simply breaking the name apart by hand. Admittedly not an elegant solution, but the alternative is probably a little complicated to code...\n\nIt would appear a relevant `LaTeX` command is the `\\titlespacing{\\subsection}` command in the `.cls` files.\n", "created_at": 1426913614, "updated_at": 1426913614}, "57410171": {"id": 57410171, "post_id": 57410171, "username": "EmmanueleSalvati", "body": "Update from original\n", "created_at": 1423709272, "updated_at": 1423709272}, "357461207": {"id": 357461207, "post_id": 357461207, "username": "brian-norman", "body": "Brian this would be significantly cooler if it auto-ran on a fixed time-period as you originally imagined... Learn how to do that! Heroku maybe?", "created_at": 1536197030, "updated_at": 1536197030}, "585591514": {"id": 585591514, "post_id": 585591514, "username": "brian-norman", "body": "", "created_at": 1584827209, "updated_at": 1584828545}, "602111660": {"id": 602111660, "post_id": 585591514, "username": "brian-norman", "body": "lol", "created_at": 1584828544, "updated_at": 1584828544}, "585583679": {"id": 585583679, "post_id": 585583679, "username": "zahin-mohammad", "body": "", "created_at": 1584824397, "updated_at": 1585023151}, "603003326": {"id": 603003326, "post_id": 585583679, "username": "zahin-mohammad", "body": "Test\r\n", "created_at": 1585023084, "updated_at": 1585023084}, "603003391": {"id": 603003391, "post_id": 585583679, "username": "zahin-mohammad", "body": "> Test\r\n\r\nYo this is a reply", "created_at": 1585023100, "updated_at": 1585023100}, "603003447": {"id": 603003447, "post_id": 585583679, "username": "zahin-mohammad", "body": "Comment 1\r\n", "created_at": 1585023113, "updated_at": 1585023113}, "603003468": {"id": 603003468, "post_id": 585583679, "username": "zahin-mohammad", "body": "Comment 2", "created_at": 1585023118, "updated_at": 1585023118}, "603003551": {"id": 603003551, "post_id": 585583679, "username": "zahin-mohammad", "body": "> Comment 1\r\n\r\nReply to 1\r\n\r\n\r\n\r\n> Comment 2\r\n\r\nReply to comment 2", "created_at": 1585023137, "updated_at": 1585023137}, "603003605": {"id": 603003605, "post_id": 585583679, "username": "zahin-mohammad", "body": "Multiline\r\nComment\r\n", "created_at": 1585023151, "updated_at": 1585023151}, "451217561": {"id": 451217561, "post_id": 451217561, "username": "brian-norman", "body": "", "created_at": 1559498415, "updated_at": 1559498415}, "356879856": {"id": 356879856, "post_id": 356879856, "username": "brian-norman", "body": "Or any user's commits that week.", "created_at": 1536077319, "updated_at": 1554088072}, "355453005": {"id": 355453005, "post_id": 355453005, "username": "brian-norman", "body": "!google \u201canything\u201d", "created_at": 1535613056, "updated_at": 1535613570}, "355452779": {"id": 355452779, "post_id": 355452779, "username": "brian-norman", "body": "!git issues user repo add \"Issue title\" \"Issue description\"", "created_at": 1535613006, "updated_at": 1535613570}, "355811190": {"id": 355811190, "post_id": 355811190, "username": "brian-norman", "body": "Glen requests this visual cue that the level has indeed changes, a good idea!\r\n", "created_at": 1535680080, "updated_at": 1535994770}, "383953069": {"id": 383953069, "post_id": 383953069, "username": "keenangaudio", "body": "", "created_at": 1543027545, "updated_at": 1543027545}, "383948347": {"id": 383948347, "post_id": 383948347, "username": "keenangaudio", "body": "", "created_at": 1543022138, "updated_at": 1543022138}, "383948004": {"id": 383948004, "post_id": 383948004, "username": "louisj381", "body": "fudge", "created_at": 1543021796, "updated_at": 1543021796}, "536209332": {"id": 536209332, "post_id": 536209332, "username": "SunnyNagam", "body": "", "created_at": 1576052051, "updated_at": 1576052051}, "592977009": {"id": 592977009, "post_id": 592977009, "username": "lewis-cooper", "body": "Is there anyway to take a graph thats already been plotted and find the gradient at each point of the graph and plot the values for the gradient on the graph?\r\n", "created_at": 1585869816, "updated_at": 1585869816}, "591643321": {"id": 591643321, "post_id": 591643321, "username": "chenxijun", "body": "To fix the bug that causes `Type Error: run() missing 1 required positional argument: 'self'`\r\n\r\nThanks for contributing to manim!\r\n\r\n**Please ensure that your pull request works with the latest version of manim.**\r\nYou should also include:\r\n\r\n1. The motivation for making this change (or link the relevant issues)\r\n2. How you tested the new behavior (e.g. a minimal working example, before/after\r\nscreenshots, gifs, commands, etc.) This is rather informal at the moment, but\r\nthe goal is to show us how you know the pull request works as intended.\r\n", "created_at": 1585719943, "updated_at": 1585895530}, "591325292": {"id": 591325292, "post_id": 591325292, "username": "applemonkey496", "body": "I just wanted to point out that pycairo can be found in [this GitHub repo](https://github.com/pygobject/pycairo), and that there it says installing it is as simple as `pip install pycairo`. Am I missing something or should this information be included in the docs? It would be much easier than locating and installing the appropriate version from [here](https://www.lfd.uci.edu/~gohlke/pythonlibs/#pycairo) as the docs say. \r\n\r\n** Also, this is my first issue, so please tell me if I'm doing something wrong.\r\n", "created_at": 1585679292, "updated_at": 1585690103}, "606804397": {"id": 606804397, "post_id": 591325292, "username": "einarf", "body": "There is more to it:\r\nhttps://pycairo.readthedocs.io/en/latest/getting_started.html\r\n\r\n> Installing Pycairo requires pkg-config and cairo including its headers\r\n\r\nThe pre-compiled versions LFD is providing is probably the easiest option on windows.", "created_at": 1585680274, "updated_at": 1585680274}, "606809639": {"id": 606809639, "post_id": 591325292, "username": "applemonkey496", "body": "@einarf, so does that mean that if I'm on MacOS (with Homebrew), to install pycairo, I would do `brew install cairo pkg-config` and then `pip3 install pycairo`? That would still be helpful to have that added to the docs, because right now I'd say the instructions for MacOS installation are quite vague.", "created_at": 1585680871, "updated_at": 1585684511}, "606886388": {"id": 606886388, "post_id": 591325292, "username": "einarf", "body": "Yeah they are. Could be good to PR doc changes in the pycairo project and link to that from Manim.", "created_at": 1585690103, "updated_at": 1585690103}, "590532375": {"id": 590532375, "post_id": 590532375, "username": "badele", "body": "Create some missing ressources for generation the all the scenes of \"from_3b1b/old/spirals\" \r\n\r\nThis can be used as a tutorials base :)", "created_at": 1585596199, "updated_at": 1585596199}, "590505279": {"id": 590505279, "post_id": 590505279, "username": "lewis-cooper", "body": "Is it possible to change the font that the axis values and the axis titles have?", "created_at": 1585593463, "updated_at": 1585593463}, "590498714": {"id": 590498714, "post_id": 590498714, "username": "kolibril13", "body": "Some GLSL compilers are more sensitive to types than others. With this fix, my Intel HD Graphics 5500 (Broadwell GT2) gives no error on this script anymore.\r\n\r\nThanks for contributing to manim!\r\n\r\n**Please ensure that your pull request works with the latest version of manim.**\r\nYou should also include:\r\n\r\n1. The motivation for making this change (or link the relevant issues)\r\n2. How you tested the new behavior (e.g. a minimal working example, before/after\r\nscreenshots, gifs, commands, etc.) This is rather informal at the moment, but\r\nthe goal is to show us how you know the pull request works as intended.\r\n", "created_at": 1585592813, "updated_at": 1585706686}, "606171037": {"id": 606171037, "post_id": 590498714, "username": "einarf", "body": "Input from a moderngl maintainer:\r\n\r\nThere might be more of this around in the glsl code. I think where floats should be expected it's typically a good idea to use `1.` or `1.0` instead of `1` as it could lead to strange behaviour with some drivers and in this case actually a compiler error.\r\n\r\nThis fix however makes Manim run at least!\r\n\r\nEDIT: Just saw this is also fixed in #943", "created_at": 1585593508, "updated_at": 1585607539}, "606251721": {"id": 606251721, "post_id": 590498714, "username": "zavden", "body": "It works! thanks.", "created_at": 1585602640, "updated_at": 1585602640}, "590497153": {"id": 590497153, "post_id": 590497153, "username": "azarzadavila", "body": "I added some examples and explanations about the content of the file geometry.py and animation.\r\nI tried to be exhaustive about the files I have inspected to write these modifications.\r\nI hosted this branch on readthedocs with links [animation](https://azarzadavila-manim.readthedocs.io/en/latest/animation.html), [geometry](https://azarzadavila-manim.readthedocs.io/en/latest/geometry.html).", "created_at": 1585592666, "updated_at": 1585592666}, "590320691": {"id": 590320691, "post_id": 590320691, "username": "lewis-cooper", "body": "Trying to plot some data points using sublime text and I get this error. Any help would be appreciated.\r\n\r\nCommand ran:\r\n```\r\nfrom manimlib.imports import *\r\n\r\n\r\n#Function to return data from .csv file to an array\r\n\r\ndef get_coords_from_csv(coronavirus_data):\r\n    import csv\r\n    coords = []\r\n    with open(f'coronavirus_data.csv', 'r') as csvFile:\r\n        reader = csv.reader(csvFile)\r\n        for row in reader:\r\n            x,y = row\r\n            coord = [float(x),float(y)]\r\n            coords.append(coord)\r\n    csvFile.close()\r\n    return coords\r\n\r\n\r\nclass GraphFromData(GraphScene):\r\n\tdef get_points_from_coords(self,coords):\r\n\t\treturn[\r\n\t\tself.coords_to_point(px,py)\r\n\t\tfor px,py in coords\r\n\t\t]\r\n\r\n\tdef get_dots_from_coords(self, coords, radius=0.1):\r\n\t\tpoints = self.get_points_from_coords(coords)\r\n\t\tdots = VGroup(*[\r\n\t\t\tDot(radius=radius).move_to([px,py,pz])\r\n\t\t\tfor px,py,pz in points\r\n\t\t\t]\r\n\t\t)\r\n\t\treturn dots\r\n\r\nclass DiscreteGraphFromSetPoints(VMobject):\r\n\tdef __init__(self, set_of_points, **kwargs):\r\n\t\tsuper().__init__(**kwargs)\r\n\t\tself.set_points_as_corners(set_of_points)\r\n\r\nclass SmoothGraphFromSetPoints(VMobject):\r\n\tdef __init__(self, set_of_points, **kwargs):\r\n\t\tsuper().__init__(**kwargs)\r\n\t\tself.set_points_smoothly(set_of_points)\r\n\r\n#Graph with set of points\r\nclass CustomGraph1(GraphFromData):\r\n\tdef construct(self):\r\n\t\tself.setup_axes()\r\n\t\tcoords = get_coords_from_csv(\"coronavirus_data\")\r\n\t\tdots = self.get_dots_from_coords(coords)\r\n\t\tself.add(dots)\r\n```\r\n\r\nThe error:\r\nTraceback (most recent call last):\r\n  File \"/Users/lewis/Documents/animation/manim/manimlib/extract_scene.py\", line 155, in main\r\n    scene = SceneClass(**scene_kwargs)\r\n  File \"/Users/lewis/Documents/animation/manim/manimlib/scene/scene.py\", line 53, in __init__\r\n    self.construct()\r\n  File \"coronavirus.py\", line 49, in construct\r\n    coords = get_coords_from_csv(\"coronavirus_data\")\r\n  File \"coronavirus.py\", line 13, in get_coords_from_csv\r\n    coord = [float(x),float(y)]\r\nValueError: could not convert string to float: '\\ufeff1'\r\n\r\nMany thanks :)", "created_at": 1585577246, "updated_at": 1585577283}, "590102894": {"id": 590102894, "post_id": 590102894, "username": "Micoael", "body": "I was trying to render a Linear transform scene to have a try at the animations.\r\nWhen I enter the from3b1b folder and eola series and began rendering files, when the following errors are shown:\r\n\r\n```\r\nPS D:\\3b1b\\manim>  python -m manim from_3b1b/old/eola/chapter3.py SimpleLinearTransformationScene -ps\r\nMedia will be written to D:\\3b1b\\manim\\media\\. You can change this behavior with the --media_dir flag.\r\n\r\n\r\n\r\nTraceback (most recent call last):\r\n  File \"D:\\3b1b\\manim\\manimlib\\extract_scene.py\", line 155, in main\r\n    scene = SceneClass(**scene_kwargs)\r\n  File \"D:\\3b1b\\manim\\manimlib\\scene\\scene.py\", line 51, in __init__\r\n    self.setup()\r\n  File \"D:\\3b1b\\manim\\manimlib\\scene\\vector_space_scene.py\", line 324, in setup\r\n    **self.background_plane_kwargs\r\n  File \"D:\\3b1b\\manim\\manimlib\\mobject\\coordinate_systems.py\", line 287, in __init__\r\n    super.__init__(**kwargs)\r\nTypeError: descriptor '__init__' of 'super' object needs an argument\r\n```\r\nLots of thanks!", "created_at": 1585557770, "updated_at": 1585706050}, "589883404": {"id": 589883404, "post_id": 589883404, "username": "timalive", "body": "Added a \"Death\" state to the SIR model, that depends on a recovery/death probability.  I have also modelled a dependency on healthcare capacity. I would appreciate permission to put a video together describing the changes made and to publish it as a supplement to your excellent SIR model video.\r\n\r\nI havent yet tested it on all the scenes, but have added it to a few key scenes. \r\n\r\nThanks for contributing to manim!\r\n\r\n**Please ensure that your pull request works with the latest version of manim.**\r\nYou should also include:\r\n\r\n1. The motivation for making this change (or link the relevant issues)\r\n2. How you tested the new behavior (e.g. a minimal working example, before/after\r\nscreenshots, gifs, commands, etc.) This is rather informal at the moment, but\r\nthe goal is to show us how you know the pull request works as intended.\r\n", "created_at": 1585515300, "updated_at": 1585517129}, "589661219": {"id": 589661219, "post_id": 589661219, "username": "Lalourche", "body": "Hi,\r\nThis is my first attempt at a pull request for manim.\r\n\r\nIt fixes some runtime errors when generating counting.py (no issue found on the subject).\r\n\r\n```\r\n    for point in self.get_template_configuration(place)\r\nTypeError: get_template_configuration() takes 1 positional argument but 2 were given\r\n\r\n    it.cycle(new_template)\r\nNameError: name 'it' is not defined\r\n```\r\n\r\nThese errors do not appear anymore after fix.\r\n[![Build Status](https://travis-ci.org/Lalourche/manim.svg?branch=fix-counting)](https://travis-ci.org/Lalourche/manim)\r\n\r\nCheers\r\nLalourche", "created_at": 1585430123, "updated_at": 1585430123}, "589568867": {"id": 589568867, "post_id": 589568867, "username": "edasubert", "body": "```\r\nfragment_shader\r\n===============\r\n0:179(24): error: `return' with wrong type int, in function `sdf' returning float\r\n```\r\n\r\nThanks for contributing to manim!\r\n\r\n**Please ensure that your pull request works with the latest version of manim.**\r\nYou should also include:\r\n\r\n1. The motivation for making this change (or link the relevant issues)\r\n2. How you tested the new behavior (e.g. a minimal working example, before/after\r\nscreenshots, gifs, commands, etc.) This is rather informal at the moment, but\r\nthe goal is to show us how you know the pull request works as intended.\r\n", "created_at": 1585398815, "updated_at": 1585398815}, "589400887": {"id": 589400887, "post_id": 589400887, "username": "mvfs314", "body": "I'd like to run the code:\r\n`numeros4 = TexMobject(\r\n\t\t\t\"\\\\mathbb{N}\",\r\n\t\t\t\"=\",\r\n\t\t\t\"\\\\{\",\r\n\t\t\t\"1\",\r\n\t\t\t\",\",\r\n\t\t\t\"2\",\r\n\t\t\t\",\",\r\n\t\t\t\"3\",\r\n\t\t\t\",\",\r\n\t\t\t\"\\\\ldots\",\r\n\t\t\t\"\\\\}\"\r\n\t\t)`\r\ncause I wish I could change the color of the braces separately. But I got this error:\r\n`Exception: Latex error converting to dvi.`\r\nI've already tried using the command `Brace()` or the ideas found at reddit, but it didn't work at all. Is there a way to make it possible?\r\n\r\nThanks!", "created_at": 1585341539, "updated_at": 1585400402}, "605434438": {"id": 605434438, "post_id": 589400887, "username": "Aathish04", "body": "Have you installed LaTeX? You also need quite a few LaTeX packages with it as well.\r\nPlease post the full traceback and we may be able to help you better.", "created_at": 1585395192, "updated_at": 1585395192}, "605440295": {"id": 605440295, "post_id": 589400887, "username": "mvfs314", "body": "I certainly have. This is the basic LaTeX code, I just wanna write the braces separately. \r\n\r\nHere the traceback:\r\n`Traceback (most recent call last):\r\n  File \"C:\\Manim\\manim-master\\manimlib\\extract_scene.py\", line 155, in main\r\n    scene = SceneClass(**scene_kwargs)\r\n  File \"C:\\Manim\\manim-master\\manimlib\\scene\\scene.py\", line 53, in __init__\r\n    self.construct()\r\n  File \"1.py\", line 170, in construct\r\n    self.numeros_naturais()\r\n  File \"1.py\", line 241, in numeros_naturais\r\n    \"\\\\}\"\r\n  File \"C:\\Manim\\manim-master\\manimlib\\mobject\\svg\\tex_mobject.py\", line 153, in __init__\r\n    self.break_up_by_substrings()\r\n  File \"C:\\Manim\\manim-master\\manimlib\\mobject\\svg\\tex_mobject.py\", line 184, in break_up_by_substrings\r\n    sub_tex_mob = SingleStringTexMobject(tex_string, **config)\r\n  File \"C:\\Manim\\manim-master\\manimlib\\mobject\\svg\\tex_mobject.py\", line 44, in __init__\r\n    self.template_tex_file_body\r\n  File \"C:\\Manim\\manim-master\\manimlib\\utils\\tex_file_writing.py\", line 19, in tex_to_svg_file\r\n    dvi_file = tex_to_dvi(tex_file)\r\n  File \"C:\\Manim\\manim-master\\manimlib\\utils\\tex_file_writing.py\", line 67, in tex_to_dvi\r\n    \"See log output above or the log file: %s\" % log_file)\r\nException: Latex error converting to dvi. See log output above or the log file: ./media\\Tex\\0447ed74899f41fb.log`\r\n\r\nThanks!", "created_at": 1585398194, "updated_at": 1585398194}, "605440741": {"id": 605440741, "post_id": 589400887, "username": "Aathish04", "body": "Hmm, the traceback points to a log file in `./media\\Tex\\0447ed74899f41fb.log`. Could you please post the contents of that as well?\n\nAlso, I've heard people saying that it isn't possible to write a single brace in LaTeX. You use Brace() for that, but you say you have tried that as well.\n\nIn any case, the contents of the log file would be mighty helpful.\n\n> On 28-Mar-2020, at 17:53, mvfs314 <notifications@github.com> wrote:\n> \n> \n> I certainly have. This is the basic LaTeX code, I just wanna write the braces separately.\n> \n> Here the traceback:\n> Traceback (most recent call last): File \"C:\\Manim\\manim-master\\manimlib\\extract_scene.py\", line 155, in main scene = SceneClass(**scene_kwargs) File \"C:\\Manim\\manim-master\\manimlib\\scene\\scene.py\", line 53, in __init__ self.construct() File \"1.py\", line 170, in construct self.numeros_naturais() File \"1.py\", line 241, in numeros_naturais \"\\\\}\" File \"C:\\Manim\\manim-master\\manimlib\\mobject\\svg\\tex_mobject.py\", line 153, in __init__ self.break_up_by_substrings() File \"C:\\Manim\\manim-master\\manimlib\\mobject\\svg\\tex_mobject.py\", line 184, in break_up_by_substrings sub_tex_mob = SingleStringTexMobject(tex_string, **config) File \"C:\\Manim\\manim-master\\manimlib\\mobject\\svg\\tex_mobject.py\", line 44, in __init__ self.template_tex_file_body File \"C:\\Manim\\manim-master\\manimlib\\utils\\tex_file_writing.py\", line 19, in tex_to_svg_file dvi_file = tex_to_dvi(tex_file) File \"C:\\Manim\\manim-master\\manimlib\\utils\\tex_file_writing.py\", line 67, in tex_to_dvi \"See log output above or the log file: %s\" % log_file) Exception: Latex error converting to dvi. See log output above or the log file: ./media\\Tex\\0447ed74899f41fb.log\n> \n> Thanks!\n> \n> \u2014\n> You are receiving this because you commented.\n> Reply to this email directly, view it on GitHub <https://github.com/3b1b/manim/issues/941#issuecomment-605440295>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AH5H6JBA2UIU6JM4LV7UW6TRJXT35ANCNFSM4LVIZ2FA>.\n> \n\n", "created_at": 1585398468, "updated_at": 1585398468}, "605441286": {"id": 605441286, "post_id": 589400887, "username": "mvfs314", "body": "Here all the log file:\r\n`This is pdfTeX, Version 3.14159265-2.6-1.40.21 (MiKTeX 2.9.7300 64-bit) (preloaded format=latex 2020.2.25)  28 MAR 2020 09:20\r\nentering extended mode\r\n**./media/Tex/0447ed74899f41fb.tex\r\n(media/Tex/0447ed74899f41fb.tex\r\nLaTeX2e <2020-02-02> patch level 3\r\nL3 programming layer <2020-02-21>\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/standalone\\standal\r\none.cls\"\r\nDocument Class: standalone 2018/03/26 v1.3a Class to compile TeX sub-files stan\r\ndalone\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/tools\\shellesc.sty\r\n\"\r\nPackage: shellesc 2019/11/08 v1.0c unified shell escape interface for LaTeX\r\nPackage shellesc Info: Restricted shell escape enabled on input line 77.\r\n)\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/iftex\\ifluatex.s\r\nty\"\r\nPackage: ifluatex 2019/10/25 v1.5 ifluatex legacy package. Use iftex instead.\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/iftex\\iftex.sty\"\r\nPackage: iftex 2019/11/07 v1.0c TeX engine tests\r\n))\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/xkeyval\\xkeyval.st\r\ny\"\r\nPackage: xkeyval 2014/12/03 v2.7a package option processing (HA)\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/xkeyval\\xkeyval.\r\ntex\"\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/xkeyval\\xkvutils\r\n.tex\"\r\n\\XKV@toks=\\toks14\r\n\\XKV@tempa@toks=\\toks15\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/xkeyval\\keyval.t\r\nex\"))\r\n\\XKV@depth=\\count167\r\nFile: xkeyval.tex 2014/12/03 v2.7a key=value parser (HA)\r\n))\r\n\\sa@internal=\\count168\r\n\\c@sapage=\\count169\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/standalone\\standal\r\none.cfg\"\r\nFile: standalone.cfg 2018/03/26 v1.3a Default configuration file for 'standalon\r\ne' class\r\n) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/base\\article.cls\r\n\"\r\nDocument Class: article 2019/12/20 v1.4l Standard LaTeX document class\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/base\\size10.clo\"\r\nFile: size10.clo 2019/12/20 v1.4l Standard LaTeX file (size option)\r\n)\r\n\\c@part=\\count170\r\n\\c@section=\\count171\r\n\\c@subsection=\\count172\r\n\\c@subsubsection=\\count173\r\n\\c@paragraph=\\count174\r\n\\c@subparagraph=\\count175\r\n\\c@figure=\\count176\r\n\\c@table=\\count177\r\n\\abovecaptionskip=\\skip47\r\n\\belowcaptionskip=\\skip48\r\n\\bibindent=\\dimen134\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\preview\\preview.sty\"\r\nPackage: preview 2017/04/24 11.91 (AUCTeX/preview-latex)\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/luatex85\\luatex8\r\n5.sty\"\r\nPackage: luatex85 2016/06/15 v1.4 pdftex aliases for luatex\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\preview\\prtightpage.def\"\r\n\\PreviewBorder=\\dimen135\r\n)\r\n\\pr@snippet=\\count178\r\n\\pr@box=\\box45\r\n\\pr@output=\\toks16\r\n)) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/babel\\babel.s\r\nty\"\r\nPackage: babel 2020/02/14 3.40 The Babel package\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/babel\\switch.def\r\n\"\r\nFile: switch.def 2020/02/14 3.40 Babel switching mechanism\r\n)\r\n*************************************\r\n* Local config file bblopts.cfg used\r\n*\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/arabi\\bblopts.cfg\"\r\nFile: bblopts.cfg 2005/09/08 v0.1 add Arabic and Farsi to \"declared\" options of\r\n babel\r\n)\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/babel-english\\engl\r\nish.ldf\"\r\nLanguage: english 2017/06/06 v3.3r English support from the babel system\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/babel\\babel.def\"\r\nFile: babel.def 2020/02/14 3.40 Babel common definitions\r\n\\babel@savecnt=\\count179\r\n\\U@D=\\dimen136\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/generic/babel\\txtbabel.d\r\nef\")\r\n\\bbl@readstream=\\read2\r\n\\bbl@dirlevel=\\count180\r\n)\r\nPackage babel Info: \\l@canadian = using hyphenrules for english\r\n(babel)             (\\language0) on input line 102.\r\nPackage babel Info: \\l@australian = using hyphenrules for ukenglish\r\n(babel)             (\\language77) on input line 105.\r\nPackage babel Info: \\l@newzealand = using hyphenrules for ukenglish\r\n(babel)             (\\language77) on input line 108.\r\n)) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/base\\inputenc.s\r\nty\"\r\nPackage: inputenc 2018/08/11 v1.3c Input encoding file\r\n\\inpenc@prehook=\\toks17\r\n\\inpenc@posthook=\\toks18\r\n) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/base\\fontenc.sty\r\n\"\r\nPackage: fontenc 2020/02/11 v2.0o Standard LaTeX package\r\n)\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsmath\\amsmath.st\r\ny\"\r\nPackage: amsmath 2020/01/20 v2.17e AMS math features\r\n\\@mathmargin=\\skip49\r\nFor additional information on amsmath, use the `?' option.\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsmath\\amstext.st\r\ny\"\r\nPackage: amstext 2000/06/29 v2.01 AMS text\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsmath\\amsgen.sty\r\n\"\r\nFile: amsgen.sty 1999/11/30 v2.0 generic functions\r\n\\@emptytoks=\\toks19\r\n\\ex@=\\dimen137\r\n)) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsmath\\amsbsy.\r\nsty\"\r\nPackage: amsbsy 1999/11/29 v1.2d Bold Symbols\r\n\\pmbraise@=\\dimen138\r\n) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsmath\\amsopn.s\r\nty\"\r\nPackage: amsopn 2016/03/08 v2.02 operator names\r\n)\r\n\\inf@bad=\\count181\r\nLaTeX Info: Redefining \\frac on input line 227.\r\n\\uproot@=\\count182\r\n\\leftroot@=\\count183\r\nLaTeX Info: Redefining \\overline on input line 389.\r\n\\classnum@=\\count184\r\n\\DOTSCASE@=\\count185\r\nLaTeX Info: Redefining \\ldots on input line 486.\r\nLaTeX Info: Redefining \\dots on input line 489.\r\nLaTeX Info: Redefining \\cdots on input line 610.\r\n\\Mathstrutbox@=\\box46\r\n\\strutbox@=\\box47\r\n\\big@size=\\dimen139\r\nLaTeX Font Info:    Redeclaring font encoding OML on input line 733.\r\nLaTeX Font Info:    Redeclaring font encoding OMS on input line 734.\r\n\\macc@depth=\\count186\r\n\\c@MaxMatrixCols=\\count187\r\n\\dotsspace@=\\muskip16\r\n\\c@parentequation=\\count188\r\n\\dspbrk@lvl=\\count189\r\n\\tag@help=\\toks20\r\n\\row@=\\count190\r\n\\column@=\\count191\r\n\\maxfields@=\\count192\r\n\\andhelp@=\\toks21\r\n\\eqnshift@=\\dimen140\r\n\\alignsep@=\\dimen141\r\n\\tagshift@=\\dimen142\r\n\\tagwidth@=\\dimen143\r\n\\totwidth@=\\dimen144\r\n\\lineht@=\\dimen145\r\n\\@envbody=\\toks22\r\n\\multlinegap=\\skip50\r\n\\multlinetaggap=\\skip51\r\n\\mathdisplay@stack=\\toks23\r\nLaTeX Info: Redefining \\[ on input line 2859.\r\nLaTeX Info: Redefining \\] on input line 2860.\r\n)\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsfonts\\amssymb.s\r\nty\"\r\nPackage: amssymb 2013/01/14 v3.01 AMS font symbols\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsfonts\\amsfonts.\r\nsty\"\r\nPackage: amsfonts 2013/01/14 v3.01 Basic AMSFonts support\r\n\\symAMSa=\\mathgroup4\r\n\\symAMSb=\\mathgroup5\r\nLaTeX Font Info:    Redeclaring math symbol \\hbar on input line 98.\r\nLaTeX Font Info:    Overwriting math alphabet `\\mathfrak' in version `bold'\r\n(Font)                  U/euf/m/n --> U/euf/b/n on input line 106.\r\n)) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\doublestroke\\dsfont.sty\"\r\nPackage: dsfont 1995/08/01 v0.1 Double stroke roman fonts\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\setspace\\setspace.sty\"\r\nPackage: setspace 2011/12/19 v6.7a set line spacing\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\tipa\\tipa.sty\"\r\nPackage: tipa 2002/08/08 TIPA version 1.1\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/base\\fontenc.sty\"\r\nPackage: fontenc 2020/02/11 v2.0o Standard LaTeX package\r\n(\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\tipa\\t3enc.def\"\r\nFile: t3enc.def 2001/12/31 T3 encoding\r\nNow handling font encoding T3 ...\r\n... no UTF-8 mapping file for font encoding T3\r\nLaTeX Font Info:    Trying to load font information for T1+cmss on input line 3\r\n57.\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/base\\t1cmss.fd\"\r\nFile: t1cmss.fd 2019/12/16 v2.5j Standard LaTeX font definitions\r\n)))) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\relsize\\relsize.sty\"\r\nPackage: relsize 2013/03/29 ver 4.1\r\n) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/base\\textcomp.st\r\ny\"\r\nPackage: textcomp 2020/02/02 v2.0n Standard LaTeX package\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\jknappen\\mathrsfs.sty\"\r\nPackage: mathrsfs 1996/01/01 Math RSFS package v1.0 (jk)\r\n\\symrsfs=\\mathgroup6\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\fundus-calligra\\calligra.sty\"\r\nPackage: calligra 2012/04/10 v1.9 LaTeX package calligra\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\wasysym\\wasysym.sty\"\r\nPackage: wasysym 2020/01/19 v2.4 Wasy-2 symbol support package\r\n\\symwasy=\\mathgroup7\r\nLaTeX Font Info:    Overwriting symbol font `wasy' in version `bold'\r\n(Font)                  U/wasy/m/n --> U/wasy/b/n on input line 93.\r\n)\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/ragged2e\\ragged2e.\r\nsty\"\r\nPackage: ragged2e 2019/07/28 v2.2 ragged2e Package (MS)\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/ms\\everysel.sty\"\r\nPackage: everysel 2011/10/28 v1.2 EverySelectfont Package (MS)\r\n)\r\n\\CenteringLeftskip=\\skip52\r\n\\RaggedLeftLeftskip=\\skip53\r\n\\RaggedRightLeftskip=\\skip54\r\n\\CenteringRightskip=\\skip55\r\n\\RaggedLeftRightskip=\\skip56\r\n\\RaggedRightRightskip=\\skip57\r\n\\CenteringParfillskip=\\skip58\r\n\\RaggedLeftParfillskip=\\skip59\r\n\\RaggedRightParfillskip=\\skip60\r\n\\JustifyingParfillskip=\\skip61\r\n\\CenteringParindent=\\skip62\r\n\\RaggedLeftParindent=\\skip63\r\n\\RaggedRightParindent=\\skip64\r\n\\JustifyingParindent=\\skip65\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\physics\\physics.sty\"\r\nPackage: physics \r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/l3packages/xparse\\\r\nxparse.sty\" (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/l3kern\r\nel\\expl3.sty\"\r\nPackage: expl3 2020-02-21 L3 programming layer (loader) \r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/l3backend\\l3backen\r\nd-dvips.def\"\r\nFile: l3backend-dvips.def 2020-02-23 L3 backend support: dvips\r\n\\l__pdf_internal_box=\\box48\r\n\\g__pdf_backend_object_int=\\count193\r\n\\l__pdf_backend_content_box=\\box49\r\n\\l__pdf_backend_model_box=\\box50\r\n\\g__pdf_backend_annotation_int=\\count194\r\n\\g__pdf_backend_link_int=\\count195\r\n\\g__pdf_backend_link_sf_int=\\count196\r\n))\r\nPackage: xparse 2020-02-14 L3 Experimental document command parser\r\n\\l__xparse_current_arg_int=\\count197\r\n\\g__xparse_grabber_int=\\count198\r\n\\l__xparse_m_args_int=\\count199\r\n\\l__xparse_v_nesting_int=\\count266\r\n)) (\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/xcolor\\xcolor.s\r\nty\"\r\nPackage: xcolor 2016/05/11 v2.12 LaTeX color extensions (UK)\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/graphics-cfg\\color\r\n.cfg\"\r\nFile: color.cfg 2016/01/02 v1.6 sample color configuration\r\n)\r\nPackage xcolor Info: Driver file: dvips.def on input line 225.\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/graphics-def\\dvips\r\n.def\"\r\nFile: dvips.def 2017/06/20 v3.1d Graphics/color driver for dvips\r\n)\r\nPackage xcolor Info: Model `cmy' substituted by `cmy0' on input line 1348.\r\nPackage xcolor Info: Model `RGB' extended on input line 1364.\r\nPackage xcolor Info: Model `HTML' substituted by `rgb' on input line 1366.\r\nPackage xcolor Info: Model `Hsb' substituted by `hsb' on input line 1367.\r\nPackage xcolor Info: Model `tHsb' substituted by `hsb' on input line 1368.\r\nPackage xcolor Info: Model `HSB' substituted by `hsb' on input line 1369.\r\nPackage xcolor Info: Model `Gray' substituted by `gray' on input line 1370.\r\nPackage xcolor Info: Model `wave' substituted by `hsb' on input line 1371.\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\microtype\\microtype.sty\"\r\nPackage: microtype 2019/11/18 v2.7d Micro-typographical refinements (RS)\r\n\\MT@toks=\\toks24\r\n\\MT@count=\\count267\r\nLaTeX Info: Redefining \\textls on input line 790.\r\n\\MT@outer@kern=\\dimen146\r\nLaTeX Info: Redefining \\textmicrotypecontext on input line 1354.\r\n\\MT@listname@count=\\count268\r\n(\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\microtype\\microtype-pdftex.def\"\r\nFile: microtype-pdftex.def 2019/11/18 v2.7d Definitions specific to pdftex (RS)\r\n\r\nLaTeX Info: Redefining \\lsstyle on input line 914.\r\nLaTeX Info: Redefining \\lslig on input line 914.\r\n\\MT@outer@space=\\skip66\r\n)\r\nPackage microtype Info: Loading configuration file microtype.cfg.\r\n(\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\microtype\\microtype.cfg\"\r\nFile: microtype.cfg 2019/11/18 v2.7d microtype main configuration file (RS)\r\n)) (C:\\Manim\\manim-master\\media\\Tex\\0447ed74899f41fb.aux)\r\n\\openout1 = `0447ed74899f41fb.aux'.\r\n\r\nLaTeX Font Info:    Checking defaults for OML/cmm/m/it on input line 24.\r\nLaTeX Font Info:    ... okay on input line 24.\r\nLaTeX Font Info:    Checking defaults for OMS/cmsy/m/n on input line 24.\r\nLaTeX Font Info:    ... okay on input line 24.\r\nLaTeX Font Info:    Checking defaults for OT1/cmr/m/n on input line 24.\r\nLaTeX Font Info:    ... okay on input line 24.\r\nLaTeX Font Info:    Checking defaults for T1/cmr/m/n on input line 24.\r\nLaTeX Font Info:    ... okay on input line 24.\r\nLaTeX Font Info:    Checking defaults for TS1/cmr/m/n on input line 24.\r\nLaTeX Font Info:    ... okay on input line 24.\r\nLaTeX Font Info:    Checking defaults for OMX/cmex/m/n on input line 24.\r\nLaTeX Font Info:    ... okay on input line 24.\r\nLaTeX Font Info:    Checking defaults for U/cmr/m/n on input line 24.\r\nLaTeX Font Info:    ... okay on input line 24.\r\nLaTeX Font Info:    Checking defaults for T3/cmr/m/n on input line 24.\r\nLaTeX Font Info:    Trying to load font information for T3+cmr on input line 24\r\n.\r\n(\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\tipa\\t3cmr.fd\"\r\nFile: t3cmr.fd 2001/12/31 TIPA font definitions\r\n)\r\nLaTeX Font Info:    ... okay on input line 24.\r\nPreview: Fontsize 10pt\r\nABD: EverySelectfont initializing macros\r\nLaTeX Info: Redefining \\selectfont on input line 24.\r\nLaTeX Info: Redefining \\microtypecontext on input line 24.\r\nPackage microtype Info: Generating DVI output.\r\nPackage microtype Info: Character protrusion enabled (level 2).\r\nPackage microtype Info: Using default protrusion set `alltext'.\r\nPackage microtype Info: No font expansion.\r\nPackage microtype Info: No adjustment of tracking.\r\nPackage microtype Info: No adjustment of interword spacing.\r\nPackage microtype Info: No adjustment of character kerning.\r\n(\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\microtype\\mt-cmr.cfg\"\r\nFile: mt-cmr.cfg 2013/05/19 v2.2 microtype config. file: Computer Modern Roman \r\n(RS)\r\n)\r\nLaTeX Font Info:    Trying to load font information for U+msa on input line 26.\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsfonts\\umsa.fd\"\r\nFile: umsa.fd 2013/01/14 v3.01 AMS symbols A\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\microtype\\mt-msa.cfg\"\r\nFile: mt-msa.cfg 2006/02/04 v1.1 microtype config. file: AMS symbols (a) (RS)\r\n)\r\nLaTeX Font Info:    Trying to load font information for U+msb on input line 26.\r\n\r\n(\"C:\\Users\\mvfs3\\AppData\\Local\\Programs\\MiKTeX 2.9\\tex/latex/amsfonts\\umsb.fd\"\r\nFile: umsb.fd 2013/01/14 v3.01 AMS symbols B\r\n) (\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\microtype\\mt-msb.cfg\"\r\nFile: mt-msb.cfg 2005/06/01 v1.0 microtype config. file: AMS symbols (b) (RS)\r\n)\r\nLaTeX Font Info:    Trying to load font information for U+rsfs on input line 26\r\n.\r\n(\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\jknappen\\ursfs.fd\"\r\nFile: ursfs.fd 1998/03/24 rsfs font definition file (jk)\r\n)\r\nLaTeX Font Info:    Trying to load font information for U+wasy on input line 26\r\n.\r\n(\"C:\\Program Files\\MiKTeX 2.9\\tex/latex\\wasysym\\uwasy.fd\"\r\nFile: uwasy.fd 2020/01/19 v2.4 Wasy-2 symbol font definitions\r\n)\r\n! Argument of \\align* has an extra }.\r\n<inserted text> \r\n                \\par \r\nl.27 \\{}\r\n         \r\nHere is how much of TeX's memory you used:\r\n 7327 strings out of 481508\r\n 111190 string characters out of 2912081\r\n 392778 words of memory out of 3000000\r\n 22471 multiletter control sequences out of 15000+200000\r\n 538995 words of font info for 43 fonts, out of 3000000 for 9000\r\n 1141 hyphenation exceptions out of 8191\r\n 43i,3n,60p,354b,212s stack positions out of 5000i,500n,10000p,200000b,50000s\r\n\r\nNo pages of output.\r\n`", "created_at": 1585398777, "updated_at": 1585398777}, "605441345": {"id": 605441345, "post_id": 589400887, "username": "mvfs314", "body": "And, about the `Braces()` it's not very useful when I want to use in a single text line :/", "created_at": 1585398807, "updated_at": 1585398807}, "605444417": {"id": 605444417, "post_id": 589400887, "username": "Aathish04", "body": "This appears to be a LaTeX specific problem. \r\n\r\nYou simply cannot use a single brace separately in LaTeX, and the way manim renders LaTeX (it uses \\align a lot), the parser keeps thinking it is seeing a single brace.\r\n\r\nFor the time being, i suggest you use two Brace()es, which is tedious, but perhaps someone more experienced with LaTeX can help you out.\r\n\r\nIf you feel like probing the problem deeper, you can modify tex_template.tex, and see if you can get it to work.\r\n\r\nAll the best ;)", "created_at": 1585400402, "updated_at": 1585400402}, "589007446": {"id": 589007446, "post_id": 589007446, "username": "nosnakeob", "body": "When I use:\r\n```\r\nt = TextMobject('aaaaa', '''a\r\n        \r\n a''', 'aaaaa')\r\n```\r\n![\u6279\u6ce8 2020-03-27 173840](https://user-images.githubusercontent.com/47769817/77742690-d5367800-7051-11ea-8324-bb71627aa300.png)\r\n\r\n\r\nbut when I use:\r\n`t = TextMobject('aaaaaaaaa', '\\n\\n', 'aaaaaaa')`\r\nor:\r\n```\r\nt = TextMobject('aaaaa', '''\r\n\r\n         ''', 'aaaaa')\r\n```\r\n![\u6279\u6ce8 2020-03-27 173932](https://user-images.githubusercontent.com/47769817/77742786-fdbe7200-7051-11ea-8470-cbc7933e64e9.png)\r\n\r\nIt's not what I want. Are there other ways to add a new line when using `TextMobject` like arrays?\r\n", "created_at": 1585302056, "updated_at": 1585304979}, "604925408": {"id": 604925408, "post_id": 589007446, "username": "qo4on", "body": "```\r\n'\\n'.join(['First_line', 'Second_line', 'Third_line'])\r\nTextMobject('First_line\\nSecond_line\\nThird_line')\r\n```", "created_at": 1585304793, "updated_at": 1585304979}, "588937811": {"id": 588937811, "post_id": 588937811, "username": "Ranoiaetep", "body": "Removed pycairo section on Directly (Windows), as its included in requirements.txt\r\nRemoved changing pycairo version number in the same section, version number included in requirements.txt is >= 1.18.1", "created_at": 1585293672, "updated_at": 1585293811}, "588714915": {"id": 588714915, "post_id": 588714915, "username": "MatejVe", "body": "I have succesfully installed all the packages required for working in manim. Problem arrises when trying to run the example_scenes.py from the console. Console reports no errors and it seems like everything went smoothly, however no video windows pop up and actually show the animation. I am working on windows 10, python 3.7.3 32bit", "created_at": 1585255317, "updated_at": 1585256023}, "604680067": {"id": 604680067, "post_id": 588714915, "username": "MatejVe", "body": "Figured it out. Intead of doing \r\npython3 manim.py example_scenes.py SquareToCircle -pl\r\nin the console, I have done\r\nmanim example_scenes.py SquareToCircle -pl\r\nand it works like it should now.\r\nStill not sure why the first thing didn't work though", "created_at": 1585256023, "updated_at": 1585256023}, "588606969": {"id": 588606969, "post_id": 588606969, "username": "qo4on", "body": "I get an error in Colab trying to run Manim from \"shaders\" branch. I see `tqdm` for each animation, that means they are successful. But then it gets into a loop and I have to stop it with the error:\r\n```\r\nTraceback (most recent call last):\r\n  File \"/usr/lib/python3.6/runpy.py\", line 193, in _run_module_as_main\r\n    \"__main__\", mod_spec)\r\n  File \"/usr/lib/python3.6/runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/content/manim/manim.py\", line 5, in <module>\r\n    manimlib.main()\r\n  File \"/content/manim/manimlib/__init__.py\", line 14, in main\r\n    scene.run()\r\n  File \"/content/manim/manimlib/scene/scene.py\", line 77, in run\r\n    self.tear_down()\r\n  File \"/content/manim/manimlib/scene/scene.py\", line 96, in tear_down\r\n    self.interact()\r\n  File \"/content/manim/manimlib/scene/scene.py\", line 104, in interact\r\n    self.update_frame()\r\n  File \"/content/manim/manimlib/scene/scene.py\", line 139, in update_frame\r\n    self.camera.clear()\r\n  File \"/content/manim/manimlib/camera/camera.py\", line 114, in clear\r\n    rgba = (*Color(self.background_color).get_rgb(), self.background_opacity)\r\n  File \"/usr/local/lib/python3.6/dist-packages/colour.py\", line 988, in __init__\r\n    self.web = color if color else 'black'\r\n  File \"/usr/local/lib/python3.6/dist-packages/colour.py\", line 1006, in __setattr__\r\n    fc(value)\r\n  File \"/usr/local/lib/python3.6/dist-packages/colour.py\", line 1084, in set_web\r\n    self.hex = web2hex(value)\r\n  File \"/usr/local/lib/python3.6/dist-packages/colour.py\", line 1005, in __setattr__\r\n    fc = getattr(self, 'set_' + label)\r\nKeyboardInterrupt\r\n```\r\n**UPDATE:\r\nIf you want to try it in Colab, [the code is here](https://github.com/moderngl/moderngl/issues/392#issuecomment-605913255).**", "created_at": 1585244760, "updated_at": 1585593681}, "605602579": {"id": 605602579, "post_id": 588606969, "username": "timalive", "body": "I am getting the same issue when running in docker. Something to do with the fact that pyglet needs a display to run. There may be a solution using Xvfb\r\n", "created_at": 1585470217, "updated_at": 1585470217}, "605608314": {"id": 605608314, "post_id": 588606969, "username": "qo4on", "body": "xvfb doesn't work either. We need to change manim `window.py` backend from `pyglet` to `headless`.\r\n```\r\nfrom moderngl_window.context.headless.window import Window as HeadlessWindow\r\n```\r\nBut I didn't managed to make it work.", "created_at": 1585473755, "updated_at": 1585473797}, "605677632": {"id": 605677632, "post_id": 588606969, "username": "t-wolfeadam", "body": "Running Win10, I couldn't get it to work when using the preview setting (-p) due to no configs for the window:\r\n```\r\nTraceback (most recent call last):\r\n  File \"C:\\Users\\Tyler Wolfe-Adam\\Documents\\Python\\manim-shaders\\manim.py\", line 5, in <module>\r\n    manimlib.main()\r\n  File \"C:\\Users\\Tyler Wolfe-Adam\\Documents\\Python\\manim-shaders\\manimlib\\__init__.py\", line 11, in main\r\n    scenes = manimlib.extract_scene.main(config)\r\n  File \"C:\\Users\\Tyler Wolfe-Adam\\Documents\\Python\\manim-shaders\\manimlib\\extract_scene.py\", line 104, in main\r\n    scenes = get_scenes_to_render(all_scene_classes, config)\r\n  File \"C:\\Users\\Tyler Wolfe-Adam\\Documents\\Python\\manim-shaders\\manimlib\\extract_scene.py\", line 72, in get_scenes_to_render\r\n    scene = scene_class(**scene_kwargs)\r\n  File \"C:\\Users\\Tyler Wolfe-Adam\\Documents\\Python\\manim-shaders\\manimlib\\scene\\scene.py\", line 44, in __init__\r\n    self.window = Window(self, **self.window_config)\r\n  File \"C:\\Users\\Tyler Wolfe-Adam\\Documents\\Python\\manim-shaders\\manimlib\\window.py\", line 21, in __init__\r\n    super().__init__(**kwargs)\r\n  File \"C:\\Program Files\\Python37\\lib\\site-packages\\moderngl_window\\context\\pyglet\\window.py\", line 54, in __init__\r\n    config=config,\r\n  File \"C:\\Program Files\\Python37\\lib\\site-packages\\pyglet\\window\\win32\\__init__.py\", line 130, in __init__\r\n    super(Win32Window, self).__init__(*args, **kwargs)\r\n  File \"C:\\Program Files\\Python37\\lib\\site-packages\\pyglet\\window\\__init__.py\", line 591, in __init__\r\n    config = screen.get_best_config(config)\r\n  File \"C:\\Program Files\\Python37\\lib\\site-packages\\pyglet\\canvas\\base.py\", line 194, in get_best_config\r\n    raise window.NoSuchConfigException()\r\npyglet.window.NoSuchConfigException\r\n```\r\nIt _does_ work when just writing to a file (-w). However, just for the shape examples. Anything with text doesn't work due to (unrelated) issues with dvisvgm not converting the dvi to svg.", "created_at": 1585505829, "updated_at": 1585505829}, "605678964": {"id": 605678964, "post_id": 588606969, "username": "qo4on", "body": "What text doesn't work? `tex_mobject` or `text_mobject`?", "created_at": 1585506426, "updated_at": 1585506426}, "605719004": {"id": 605719004, "post_id": 588606969, "username": "t-wolfeadam", "body": "> What text doesn't work? `tex_mobject` or `text_mobject`?\r\n\r\nBoth, unfortunately; I also tried modifying the example scenes with just using one or the other. Both result in `OSError: No file matching ./media\\Tex\\ab47de5016aed0d4.svg in image directory`.\r\n\r\nI dug into it and it stems from the `dvisvgm` command (called from the `dvi_to_svg` function inside `\\manimlib\\utils\\tex_file_writing.py`) - the command executes successfully but nothing gets outputted (no files or messages, even with the highest level of verbosity). I even tried searching my entire file system for the file and still nothing. It's really weird. \r\n\r\nI'm currently getting it setup to try on WSL. Hopefully that works better. No luck on googling the problem either. I'll probably open a new thread somewhere if it doesn't work with WSL :/\r\n\r\nEdit: The problem with dvisvgm seems to be a Windows thing. I found out it's returning exit code -1073740791, which is apparently a stack buffer overflow. dvisvgm works fine on WSL, so for now I've implemented a hacky solution where Windows waits until Ubuntu has converted the necessary files. \r\n\r\nEdit 2: **Found the fix!** For future readers: Running `dvisvgm --version` reported 2.7. I ran `where dvisvgm` and it reported two versions: the v2.7 (installed in Program Files) showed before the other, which was part of MikTex and version 2.8.2. Removing the v2.7 copy fixed this issue. \r\nI am now able to render videos in this branch, as long as I'm writing to file (`w` flag) and not showing preview (`p` flag). ", "created_at": 1585524740, "updated_at": 1585588715}, "605728064": {"id": 605728064, "post_id": 588606969, "username": "t-wolfeadam", "body": "Update: Ran into the same problems with the display. Tried using xvfb like so:\r\n`xvfb-run -s \"-screen 0 640x360x24\" python3.7 -m manim example_scenes.py SquareToCircle -ql`\r\nThat returned the same error for me as when I tried to use the `q` flag on Windows:\r\n```\r\nTraceback (most recent call last):\r\n  File \"/usr/lib/python3.7/runpy.py\", line 193, in _run_module_as_main\r\n    \"__main__\", mod_spec)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/home/tyler/Documents/manim/manim.py\", line 5, in <module>\r\n    manimlib.main()\r\n  File \"/home/tyler/Documents/manim/manimlib/__init__.py\", line 11, in main\r\n    scenes = manimlib.extract_scene.main(config)\r\n  File \"/home/tyler/Documents/manim/manimlib/extract_scene.py\", line 104, in main\r\n    scenes = get_scenes_to_render(all_scene_classes, config)\r\n  File \"/home/tyler/Documents/manim/manimlib/extract_scene.py\", line 72, in get_scenes_to_render\r\n    scene = scene_class(**scene_kwargs)\r\n  File \"/home/tyler/Documents/manim/manimlib/scene/scene.py\", line 44, in __init__\r\n    self.window = Window(self, **self.window_config)\r\n  File \"/home/tyler/Documents/manim/manimlib/window.py\", line 21, in __init__\r\n    super().__init__(**kwargs)\r\n  File \"/home/tyler/.local/lib/python3.7/site-packages/moderngl_window/context/pyglet/window.py\", line 54, in __init__\r\n    config=config,\r\n  File \"/home/tyler/.local/lib/python3.7/site-packages/pyglet/window/xlib/__init__.py\", line 171, in __init__\r\n    super(XlibWindow, self).__init__(*args, **kwargs)\r\n  File \"/home/tyler/.local/lib/python3.7/site-packages/pyglet/window/__init__.py\", line 591, in __init__\r\n    config = screen.get_best_config(config)\r\n  File \"/home/tyler/.local/lib/python3.7/site-packages/pyglet/canvas/base.py\", line 194, in get_best_config\r\n    raise window.NoSuchConfigException()\r\npyglet.window.NoSuchConfigException\r\n```\r\n\r\nTried again with the `w` flag, but then got the following error:\r\n```\r\nMedia will be written to ./media/. You can change this behavior with the --media_dir flag.\r\nTraceback (most recent call last):\r\n  File \"/usr/lib/python3.7/runpy.py\", line 193, in _run_module_as_main\r\n    \"__main__\", mod_spec)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/home/tyler/Documents/manim/manim.py\", line 5, in <module>\r\n    manimlib.main()\r\n  File \"/home/tyler/Documents/manim/manimlib/__init__.py\", line 14, in main\r\n    scene.run()\r\n  File \"/home/tyler/Documents/manim/manimlib/scene/scene.py\", line 74, in run\r\n    self.construct()\r\n  File \"example_scenes.py\", line 84, in construct\r\n    self.play(Transform(square, circle))\r\n  File \"/home/tyler/Documents/manim/manimlib/scene/scene.py\", line 357, in wrapper\r\n    func(self, *args, **kwargs)\r\n  File \"/home/tyler/Documents/manim/manimlib/scene/scene.py\", line 425, in play\r\n    self.progress_through_animations(animations)\r\n  File \"/home/tyler/Documents/manim/manimlib/scene/scene.py\", line 402, in progress_through_animations\r\n    self.update_frame(dt)\r\n  File \"/home/tyler/Documents/manim/manimlib/scene/scene.py\", line 140, in update_frame\r\n    self.camera.capture(*self.mobjects)\r\n  File \"/home/tyler/Documents/manim/manimlib/camera/camera.py\", line 228, in capture\r\n    shader = self.get_shader(info_group[0])\r\n  File \"/home/tyler/Documents/manim/manimlib/camera/camera.py\", line 253, in get_shader\r\n    fragment_shader=get_shader_code_from_file(info[\"frag\"]),\r\n  File \"/home/tyler/.local/lib/python3.7/site-packages/moderngl/context.py\", line 1037, in program\r\n    varyings\r\nmoderngl.error.Error: GLSL Compiler failed\r\n\r\nfragment_shader\r\n===============\r\n0:179(24): error: `return' with wrong type int, in function `sdf' returning float\r\n``", "created_at": 1585528748, "updated_at": 1585528748}, "605798041": {"id": 605798041, "post_id": 588606969, "username": "qo4on", "body": "You can't get hardware acceleration with virtual displays such as xvfb. Can you render `SquareToCircle` in `shaders` version correctly?", "created_at": 1585547883, "updated_at": 1585547883}, "605830554": {"id": 605830554, "post_id": 588606969, "username": "timalive", "body": "I was able to do it on mac - not in docker ", "created_at": 1585553299, "updated_at": 1585553299}, "605846659": {"id": 605846659, "post_id": 588606969, "username": "qo4on", "body": "Can you try to run it in Colab? I had no success there.", "created_at": 1585555532, "updated_at": 1585555532}, "605947824": {"id": 605947824, "post_id": 588606969, "username": "einarf", "body": "Lengthy issue about this in the modengl project: https://github.com/moderngl/moderngl/issues/392\r\nWe do actually get it running in Google Colab with EGL, but it gets stuck in an infinite loop. That I think is a flow issue in the Manim code itself.. or at least it needs to be debugged properly to make a conclusion.\r\n\r\nManim is using moderngl and moderngl-window. I think the main issues is that it extends the pyglet window backend directly. If headless rendering is needed it should use the `headless.Window` instead.\r\n\r\nThe `BaseWindow.init_mgl_context` should be probably also be overridden to create the correct context type or I can add some features for this in the moderngl-window project if we figure out what is needed. Right now it tries to **detect** the context pyglet creates**, but in headless we just need to make our own context with no window (X11 or EGL).\r\n\r\nIt's not that complicated to fix thankfully.\r\n\r\nI am willing to help out giving pointers for the moderngl specific things, but I don't really know much about Manim (at least yet).", "created_at": 1585568204, "updated_at": 1585569436}, "606115199": {"id": 606115199, "post_id": 588606969, "username": "t-wolfeadam", "body": "> You can't get hardware acceleration with virtual displays such as xvfb. Can you render `SquareToCircle` in `shaders` version correctly?\r\n\r\nGood to know! I can in Windows, but not in the Ubuntu subsystem. For both the `q` and `w` flags I get the `pyglet.canvas.xlib.NoSuchDisplayException: Cannot connect to \"None\"` exception.", "created_at": 1585587005, "updated_at": 1585587005}, "606136266": {"id": 606136266, "post_id": 588606969, "username": "einarf", "body": "I released a new version of moderngl-window what should at least solve the `NoSuchConfig` for a lot of people. You maye also have to go into the `window.py` module and set `samples = 0`.\r\n\r\n@t-wolfeadam The shader error you are getting is an error in the glsl code here : https://github.com/3b1b/manim/blob/2ce0b72c440fadbfba990bdac8bc9c2ea8bff4e7/manimlib/shaders/quadratic_bezier_fill_frag.glsl#L41\r\n\r\nIt should return a `float`, not an `int`. Not all GLSL compilers are nice and convert constants for you automatically. I would fix that line and test it again. Make a PR as well while you are at it (if it works) \ud83d\ude04 \r\n\r\nYou may see more of those, but they are quick and easy to fix if you search for the variable or function names reported by the GLSL compiler.\r\n", "created_at": 1585589418, "updated_at": 1585589554}, "606164014": {"id": 606164014, "post_id": 588606969, "username": "t-wolfeadam", "body": "> I released a new version of moderngl-window what should at least solve the `NoSuchConfig` for a lot of people. You may also have to go into the `window.py` module and set `samples = 0`.\r\n> \r\n\r\nFantastic, thanks! I can confirm for Windows that updating + setting samples to 0 fixed it!\r\n\r\n> @t-wolfeadam The shader error you are getting is an error in the glsl code here :\r\n> \r\n> https://github.com/3b1b/manim/blob/2ce0b72c440fadbfba990bdac8bc9c2ea8bff4e7/manimlib/shaders/quadratic_bezier_fill_frag.glsl#L41\r\n> \r\n> It should return a `float`, not an `int`. Not all GLSL compilers are nice and convert constants for you automatically. I would fix that line and test it again. Make a PR as well while you are at it (if it works) \ud83d\ude04\r\n\r\nAhh, so that's what sdf was referring to. That also did the trick, so thank you for that as well! (and fortunately, it was the only instance).\r\n\r\nEdit: looks like someone already made the PR for that issue", "created_at": 1585592705, "updated_at": 1585592791}, "606171934": {"id": 606171934, "post_id": 588606969, "username": "einarf", "body": "@t-wolfeadam Yep. We just sorted out a PR after discussing the issue on the manim discord server \ud83d\udc4d ", "created_at": 1585593620, "updated_at": 1585593620}, "606172498": {"id": 606172498, "post_id": 588606969, "username": "kolibril13", "body": "And here it is:\r\nhttps://github.com/3b1b/manim/pull/950", "created_at": 1585593681, "updated_at": 1585593681}, "588343035": {"id": 588343035, "post_id": 588343035, "username": "girishgargcool", "body": "Many people face problems while installing manim. This is partly due to the fact that is isn't well documented. I myself had come across so many problems setting up manim. \r\n\r\nSo I have added a .ova file which will help to install manim in a virtual Ubuntu os on the host os. All the requirements and dependencies are pre-installed and the user can directly start making beautiful animations.\r\n\r\nCredits: This .ova file was made by [Purusharth Saxena](https://github.com/purusharths)", "created_at": 1585221453, "updated_at": 1585221453}, "587571726": {"id": 587571726, "post_id": 587571726, "username": "Enempie", "body": "Hello everyone,\r\nI'm pretty new to Python and neither had to set up different environments nor install a package (or something for it) outside of Anaconda or without the simple conda install <package> command.\r\nAs stated, I use Anaconda 3 and usually work with Spyder 4 on Windows 10 Pro 64 Bit. \r\nI somehow successfully installed the manim package and set up a manim environment by partially following a YT video and the instructions on this git. \r\nI have the following problems:\r\n1. I can't setup a terminal in Spyder with my manim env. I guess Spyder can't find my virtualenv?\r\n2. I can choose the manim env in VSCode and successfully render the example_scenes but only if I go to my manim directory from the terminal first which is okay but I don't know if that's the only way or if I set up something wrong (I'm not used to be forced to do that, is there a way to import manim and call the methods like I'm used to independent of the directory of my terminal)? I'm just a simple engineer and I honestly have no clue how complex software works in the background, I only know how to run my math stuff..As I said it's currently working (just from VSCode) but I'm wondering if it is possible to get it to my usual workflow (running the manim env in Spyder and especially calling manim methods without going to the directory with cd \\.. commands first).\r\nI'd appreciate a helping hand a lot and thanks a lot to this amazing community in advance.\r\n\r\nSincerely ", "created_at": 1585129861, "updated_at": 1585154891}, "603954587": {"id": 603954587, "post_id": 587571726, "username": "Enempie", "body": "Okay, \r\nI managed to pin it down a bit. I got Anaconda to recognize my manim env and got it to the Spyder terminal. What's now still bothering me is, that I don't seem to be able to change directory in the IPython console of Spyder and therefore can't run the extraction. Again, it works perfectly in VSCode but I'd love to have the Spyder variable explorer for my purposes.\r\nI'll post the console log asap. ", "created_at": 1585154891, "updated_at": 1585154891}, "585766275": {"id": 585766275, "post_id": 585766275, "username": "Steff-san", "body": "I tried to install manin for linux but i get this error:\r\nBuilding wheels for collected packages: pycairo\r\n  Building wheel for pycairo (setup.py) ... error\r\n  ERROR: Command errored out with exit status 1:\r\n   command: /usr/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-95eiqbht/pycairo/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-95eiqbht/pycairo/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d /tmp/pip-wheel-vgeh7_q5\r\n       cwd: /tmp/pip-install-95eiqbht/pycairo/\r\n  Complete output (17 lines):\r\n  running bdist_wheel\r\n  running build\r\n  running build_py\r\n  creating build\r\n  creating build/lib.linux-x86_64-3.7\r\n  creating build/lib.linux-x86_64-3.7/cairo\r\n  copying cairo/__init__.py -> build/lib.linux-x86_64-3.7/cairo\r\n  running build_ext\r\n  building 'cairo._cairo' extension\r\n  creating build/temp.linux-x86_64-3.7\r\n  creating build/temp.linux-x86_64-3.7/cairo\r\n  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DPYCAIRO_VERSION_MAJOR=1 -DPYCAIRO_VERSION_MINOR=16 -DPYCAIRO_VERSION_MICRO=1 -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/python3.7m -c cairo/device.c -o build/temp.linux-x86_64-3.7/cairo/device.o\r\n  cairo/device.c:30:10: fatal error: Python.h: No existe el archivo o el directorio\r\n   #include <Python.h>\r\n            ^~~~~~~~~~\r\n  compilation terminated.\r\n  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1\r\n  ----------------------------------------\r\n  ERROR: Failed building wheel for pycairo\r\n  Running setup.py clean for pycairo\r\nFailed to build pycairo\r\nInstalling collected packages: pycairo\r\n  Attempting uninstall: pycairo\r\n    Found existing installation: pycairo 1.16.2\r\nERROR: Cannot uninstall 'pycairo'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.\r\n\r\nNote: \r\nIn windous i have got a similar errror.", "created_at": 1584896980, "updated_at": 1585257497}, "603367609": {"id": 603367609, "post_id": 585766275, "username": "Aathish04", "body": "I think you're new to Github.\r\nIt's considered bad practice to have an uninformative title and issue.\r\nWe need a bit more info from you. What OS are you using, and what were you trying to do?", "created_at": 1585068985, "updated_at": 1585068985}, "604357553": {"id": 604357553, "post_id": 585766275, "username": "Nikhil0504", "body": "> I think you're new to Github.\r\n> It's considered bad practice to have an uninformative title and issue.\r\n> We need a bit more info from you. What OS are you using, and what were you trying to do?\r\n\r\nhe is using linux and was trying download the repo requirements ", "created_at": 1585219508, "updated_at": 1585219508}, "584628633": {"id": 584628633, "post_id": 584628633, "username": "luketimmerman314", "body": "So, I have my own little creature that I made (it is an SVG file) using InkScape. However, it was not rendering correctly when I ran my python code. For reference, I have attached the error message that I am receiving when I try and run my code:\r\n\r\nMedia will be written to ./media/. You can change this behavior with the --media_dir flag.\r\nconversation.py:39: UserWarning: No PsiCreatures design with mode happy\r\n  (self.file_name_prefix, mode))\r\n\r\n\r\n\r\nTraceback (most recent call last):\r\n  File \"conversation.py\", line 36, in __init__\r\n    SVGMobject.__init__(self, file_name=svg_file, **kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 46, in __init__\r\n    VMobject.__init__(self, **kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/mobject.py\", line 48, in __init__\r\n    self.generate_points()\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 69, in generate_points\r\n    mobjects = self.get_mobjects_from(svg)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 87, in get_mobjects_from\r\n    for child in element.childNodes\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 87, in <listcomp>\r\n    for child in element.childNodes\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 87, in get_mobjects_from\r\n    for child in element.childNodes\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 87, in <listcomp>\r\n    for child in element.childNodes\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 91, in get_mobjects_from\r\n    element.getAttribute('d')\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 119, in path_string_to_mobject\r\n    return VMobjectFromSVGPathstring(path_string)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 322, in __init__\r\n    VMobject.__init__(self, **kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/mobject.py\", line 48, in __init__\r\n    self.generate_points()\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 349, in generate_points\r\n    self.handle_command(command, coord_string)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 359, in handle_command\r\n    new_points = self.string_to_points(coord_string)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 422, in string_to_points\r\n    numbers = string_to_numbers(coord_string)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 24, in string_to_numbers\r\n    for s in re.split(\"[ ,]\", num_string)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 25, in <listcomp>\r\n    if s != \"\"\r\nValueError: could not convert string to float: '6.39.22'\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/extract_scene.py\", line 155, in main\r\n    scene = SceneClass(**scene_kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/scene/scene.py\", line 53, in __init__\r\n    self.construct()\r\n  File \"conversation.py\", line 70, in construct\r\n    psi_creature = PsiCreature()\r\n  File \"conversation.py\", line 44, in __init__\r\n    SVGMobject.__init__(self, mode=\"happy\", file_name=svg_file, **kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 45, in __init__\r\n    self.ensure_valid_file()\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 63, in ensure_valid_file\r\n    self.file_name)\r\nOSError: No file matching assets/PsiCreatures_happy.svg in image directory\r\n\r\nThanks for any help y'all can give!", "created_at": 1584644123, "updated_at": 1585291678}, "602088618": {"id": 602088618, "post_id": 584628633, "username": "luketimmerman314", "body": "Hello?", "created_at": 1584817584, "updated_at": 1584817584}, "602088666": {"id": 602088666, "post_id": 584628633, "username": "luketimmerman314", "body": "Do y'all need more information?\r\n", "created_at": 1584817602, "updated_at": 1584817602}, "602119164": {"id": 602119164, "post_id": 584628633, "username": "luketimmerman314", "body": "So, I've been experimenting today, and I was able to get some of the simple .svg files I created in InkScape to render correctly when I ran my code. But, as the images got more complex, the images rendered completely terribly. I don't understand why this would be. I'm using the same application, and I was even able to get Grant's PiCreatures_plain.svg to run through InkScape. Any suggestions?", "created_at": 1584833447, "updated_at": 1584833447}, "603591323": {"id": 603591323, "post_id": 584628633, "username": "raghavg123", "body": "The default directory for manim to look for svg images is `assets/svg_images` (outside the manimlib folder) so try putting your file `PsiCreatures_happy.svg` in that folder (if it doesn't exist, create it), so that should solve your second problem.", "created_at": 1585100099, "updated_at": 1585100099}, "604842764": {"id": 604842764, "post_id": 584628633, "username": "chiraagChakravarthy", "body": "Hey. I'm having a similar issue, where my ellipses and rectangles render fine, but the lines just don't show up. Do you have any insight into why that may happen?", "created_at": 1585291678, "updated_at": 1585291678}, "584556456": {"id": 584556456, "post_id": 584556456, "username": "Haffi112", "body": "Hi everyone,\r\n\r\nI am new to manim so please bear with me. I am trying to run older scenes and for them I am unable to run scenes which involve pi creatures. Is there a standard workaround for this?\r\n\r\nIn particular I get the following error:\r\n\r\n    UserWarning: No PiCreatures design with mode plain (self.file_name_prefix, mode))\r\n\r\nwhen running the following command from the manim directory:\r\n\r\n    python manim.py from_3b1b/old/nn/part1.py -pl\r\n\r\nAs an example I chose scene number 28.\r\n\r\nAny help would be appreciated. In case the images are not available I would be happy with a placeholder just to see and confirm that things work.", "created_at": 1584636834, "updated_at": 1584639752}, "601320562": {"id": 601320562, "post_id": 584556456, "username": "Haffi112", "body": "I tried fixing it myself and noticed a couple of problems.\r\n\r\nImage assets for creatures were missing. I found some [here](https://www.dropbox.com/sh/fybvhyyr2mls40v/AABv9hQPuS_dJUhdKYdMJ3LWa/designs/PiCreature?dl=0&lst=&subfolder_nav_tracking=1) after reading [this](https://www.reddit.com/r/3Blue1Brown/comments/8v247s/i_have_been_inspired_by_3blue1brown_for_a_while/) Reddit thread. Note that these are not the standard creatures from the 3b1b youtube channel but it was never my goal to replicate that anyway.\r\n\r\nThe next hurdle is the MEDIA_DIR variable. It is by default set by the constants.py file but in the pi_creature.py file it shows up as an empty string. For the sake of making things work I decided to change it locally in the pi_creature.py file but I assume that is not the intended solution.\r\n\r\nIt would be great if there were some default assets to replace the normal pi_creatures and a fix for the above problem. Besides these issues I am really happy with manim ^^", "created_at": 1584639736, "updated_at": 1584639752}, "583833153": {"id": 583833153, "post_id": 583833153, "username": "hanwenzhu", "body": "I worked on manim back in 2018. I added the function for using CTeX (XeLaTeX package for Chinese) and XeLaTeX instead of LaTeX using the flag `TEX_USE_CTEX` in constants.py (#315).\r\n\r\nI have stopped working on manim since 2019, but over the months there are apparently more and more people who want to use LaTeX rendering in non-English languages, and even on very old issues I still occasionally see people asking how to do that... Looking back at my change I really should have **decoupled using CTeX (TeX template) from XeLaTeX (rendering tool)**. This has caused a *lot* of confusions and made weird hacks/fixes necessary for only using XeLaTeX, especially for a language that is not Chinese or English, with the most recent #858 and #840. It really should have been a flag `TEX_USE_XELATEX` and another flag `TEMPLATE_TEX_NAME`, and the flag `TEX_USE_CTEX` is such that when it is `True`, `TEX_USE_XELATEX` is `True` and `TEMPLATE_TEX_NAME` is `\"ctex_template.tex\"`; otherwise `TEX_USE_XELATEX` is `False` and `TEMPLATE_TEX_NAME` is `\"tex_template.tex\"`. Then set `TEMPLATE_TEX_FILE` to `os.path.join(os.path.dirname(os.path.realpath(__file__)), TEMPLATE_TEX_NAME)`. Corresponding logic: constants.py lines 74\u201379.\r\n\r\nIt might be even better to set it dynamically using a function or as a parameter of `TexMobject()`, (see issues like #891). I looked at the source code and this is definitely possible. The options I can think of are\r\n1. Use the current `TEX_USE_CTEX`\r\n2. Add flags `TEX_USE_XELATEX` and `TEMPLATE_TEX_NAME`, and rework `TEX_USE_CTEX`\r\n3. Add parameters for `TexMobject()` like `use_xelatex=False` and `tex_template=\"tex_template.tex\"`\r\n4. Use the flags of 2. as a default, and make it possible to change the default using 3.\r\n\r\nNot really sure if this is the right place to raise this issue.\r\n", "created_at": 1584547113, "updated_at": 1584585223}, "600722090": {"id": 600722090, "post_id": 583833153, "username": "hanwenzhu", "body": "I can do a PR for the change if needed", "created_at": 1584547879, "updated_at": 1584547879}, "600955502": {"id": 600955502, "post_id": 583833153, "username": "hanwenzhu", "body": "#723 is also a pretty good idea and there is a similar pr #745 although it doesn't decouple CTeX from XeLaTeX", "created_at": 1584585223, "updated_at": 1584585223}, "583286873": {"id": 583286873, "post_id": 583286873, "username": "luketimmerman314", "body": "Hello, I am having trouble getting an .svg image to work in the newest version of manim. Any help would be appreciated. I have attached a screen shot of the code I am using, and the error messages that I am getting. In making my own code, I tried to mimic the PiCreatures code as close as possible to get it to run but to no avail.\r\n\r\n![Screen Shot 2020-03-17 at 3 41 03 PM](https://user-images.githubusercontent.com/62107786/76899643-f6cf7c80-6865-11ea-87ac-aaeebb6d7539.png)\r\n![Screen Shot 2020-03-17 at 3 41 21 PM](https://user-images.githubusercontent.com/62107786/76899650-fa630380-6865-11ea-92e5-ed16b4a34b80.png)\r\n![Screen Shot 2020-03-17 at 3 41 42 PM](https://user-images.githubusercontent.com/62107786/76899659-fd5df400-6865-11ea-8759-2c515fc4a91a.png)\r\n![Screen Shot 2020-03-17 at 3 41 58 PM](https://user-images.githubusercontent.com/62107786/76899663-ff27b780-6865-11ea-8376-c738552b8231.png)\r\n", "created_at": 1584477819, "updated_at": 1584637040}, "600451490": {"id": 600451490, "post_id": 583286873, "username": "mental32", "body": "This doesn't appear to be an issue with the library, at a guess you're missing a `FILE_DIR` that appears to be either `PSI_CREATURE_DIR` or `MEDIA_DIR`.\r\n\r\nAlso for future refrence it's better to attach a link to a [gist](https://gist.github.com/) or [hastebin](https://hastebin.com) with the relevant code and tracebacks instead of screenshots.", "created_at": 1584513733, "updated_at": 1584515438}, "600646029": {"id": 600646029, "post_id": 583286873, "username": "luketimmerman314", "body": "So how would I remedy this situation? How do I create a PSI_CREATURE_DIR or a MEDIA_DIR?", "created_at": 1584540748, "updated_at": 1584540748}, "600705024": {"id": 600705024, "post_id": 583286873, "username": "luketimmerman314", "body": "Also, where is the image directory?\r\n", "created_at": 1584546533, "updated_at": 1584546533}, "600773437": {"id": 600773437, "post_id": 583286873, "username": "mental32", "body": "> So how would I remedy this situation? How do I create a PSI_CREATURE_DIR or a MEDIA_DIR?\r\n\r\nBoth of those names are defined already at the top of the file so it's as simple as replacing the undefined name with either one of those.\r\n\r\nAlso for getting started with Python I'd recommend the [official tutorial](https://docs.python.org/3/tutorial/) or the [realpython blog](https://realpython.com/) and for asking questions there's [r/learnpython](https://www.reddit.com/r/learnpython/), one of the [many](https://discordapp.com/invite/python) [discord](https://discord.gg/RAKc3HF) [guilds](https://discordapp.com/invite/code) or the #python channel on the [freenode IRC](https://freenode.net/).\r\n", "created_at": 1584553694, "updated_at": 1584553694}, "600825398": {"id": 600825398, "post_id": 583286873, "username": "luketimmerman314", "body": "Ok, so I changed the FILE_DIR to PSI_CREATURE_DIR, and that got rid of the FILE_DIR Not Defined Error. Thank you for that.\r\n\r\nBut it still says \r\nOSError: No file matching assets/PsiCreatures_happy.svg in image directory\r\n\r\nDoes this have something to do with where I have placed the svg file? Currently, I have placed the svg file in manim/manimlib/files, as this is where the PiCreatures_plain.svg also resides. I tried moving the svg to the assets file to see if that would work, but it still returns the same error message. Any advice for this?", "created_at": 1584560807, "updated_at": 1584560807}, "600948199": {"id": 600948199, "post_id": 583286873, "username": "hanwenzhu", "body": "`FILE_DIR` was defined way back to be os.path.join(\".\", \"files\") and is removed in commit caa4577cd161e5c8bd4f2f57b9d6c3295672d2a3. Add this line to the top of your script\r\n```py\r\nFILE_DIR = os.path.join(\".\", \"files\")\r\n```\r\nafter the `import *` if your SVG files are placed in ./files\r\n\r\nIt seems that the FILE_DIR reference is part of the error handling block. Check if the SVG files are in ./media/assets or ./assets first", "created_at": 1584583536, "updated_at": 1584584006}, "600960816": {"id": 600960816, "post_id": 583286873, "username": "luketimmerman314", "body": "Ok, so I moved the file PsiCreatures_happy.svg back to the files folder in the library and then added the code you provided. However, it is still giving me this error when I am trying to run it in Terminal:\r\n\r\nMedia will be written to ./media/. You can change this behavior with the --media_dir flag.\r\nconversation.py:39: UserWarning: No PsiCreatures design with mode happy\r\n  (self.file_name_prefix, mode))\r\n\r\n\r\n\r\nTraceback (most recent call last):\r\n  File \"conversation.py\", line 36, in __init__\r\n    SVGMobject.__init__(self, file_name=svg_file, **kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 45, in __init__\r\n    self.ensure_valid_file()\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 63, in ensure_valid_file\r\n    self.file_name)\r\nOSError: No file matching files/PsiCreatures_happy.svg in image directory\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/extract_scene.py\", line 155, in main\r\n    scene = SceneClass(**scene_kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/scene/scene.py\", line 53, in __init__\r\n    self.construct()\r\n  File \"conversation.py\", line 70, in construct\r\n    psi_creature = PsiCreature()\r\n  File \"conversation.py\", line 44, in __init__\r\n    SVGMobject.__init__(self, mode=\"happy\", file_name=svg_file, **kwargs)\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 45, in __init__\r\n    self.ensure_valid_file()\r\n  File \"/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py\", line 63, in ensure_valid_file\r\n    self.file_name)\r\nOSError: No file matching assets/PsiCreatures_happy.svg in image directory\r\n\r\nAny ideas? Does it have something to do with the fact that it is trying to work with the assets folder? Because this folder is no longer where those svg images are stored.\r\n\r\nThanks!", "created_at": 1584586521, "updated_at": 1584586521}, "600976923": {"id": 600976923, "post_id": 583286873, "username": "hanwenzhu", "body": "The `files` folder is meant to be in the directory that you called your python program (presumably the same directory as `conversation.py`", "created_at": 1584590586, "updated_at": 1584590586}, "600979597": {"id": 600979597, "post_id": 583286873, "username": "luketimmerman314", "body": "Ok, thank you so much! That got it to play, but now there is a different issue: it looks nothing like the image that I made using the InkScape program. Any suggestions on how to fix that? I tried using it with Grant's Pi Creatures, and that worked perfectly, so it must be something with my file or code, maybe?\r\n\r\nThanks!", "created_at": 1584591345, "updated_at": 1584591345}, "600989330": {"id": 600989330, "post_id": 583286873, "username": "hanwenzhu", "body": "I'm not sure, but my guess is that it's manim's problem working with SVG files, since if I remember correctly manim uses its own SVG parsing code. This works fine on LaTeX-generated SVG but most likely not on any SVG.", "created_at": 1584593969, "updated_at": 1584594156}, "600994069": {"id": 600994069, "post_id": 583286873, "username": "luketimmerman314", "body": "Ok thanks for the clarification. Is there somewhere you could point me that would show me how to generate my svg file using LaTex? Or perhaps somehow convert my .svg to be compatible with the manim code?\r\n\r\nThanks again for all of your help! I really appreciate it!", "created_at": 1584595246, "updated_at": 1584595246}, "601005602": {"id": 601005602, "post_id": 583286873, "username": "hanwenzhu", "body": "The .tex file generates a .dvi file using the `latex` command, which is used to generate the .svg file using `dvisvgm`. This is in [tex_file_writing.py](https://github.com/3b1b/manim/blob/master/manimlib/utils/tex_file_writing.py). The SVG parsing part is at the definition of SVGMobject [here](https://github.com/3b1b/manim/blob/master/manimlib/mobject/svg/svg_mobject.py#L76). It only handles `<defs>`, `<g>/<svg>/<symbol>`, `<path>`, `<use>`, `<rect>`, `<circle>`, `<ellipse>`, and `<polygon>/<polyline>`.", "created_at": 1584598118, "updated_at": 1584598118}, "601012984": {"id": 601012984, "post_id": 583286873, "username": "luketimmerman314", "body": "All that my PsiCreature is built from are paths and ellipses, so I'm not sure if that's the problem. So I found the image that Grant uses (PiCreatures_plain.svg) and opened it with just the standard TextEdit Program. It looks as if Grant used Adobe Illustrator to construct the image, and then used an SVG Export plug In to export the file. Do you think that if I rebuilt the image using Illustrator instead of InkScape that it would render correctly? For reference, I have attached the code I received from PiCreatures_plain.svg\r\n\r\n<?xml version=\"1.0\" encoding=\"utf-8\"?>\r\n<!-- Generator: Adobe Illustrator 21.0.2, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->\r\n<svg version=\"1.1\" id=\"Layer_1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" x=\"0px\" y=\"0px\"\r\n\t viewBox=\"0 0 480 480\" style=\"enable-background:new 0 0 480 480;\" xml:space=\"preserve\">\r\n<style type=\"text/css\">\r\n\t.st0{fill:#FFFFFF;}\r\n\t.st1{fill:#0C7F99;}\r\n</style>\r\n<path class=\"st0\" d=\"M251.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19\r\n\tC243.1,95.6,251.6,104.2,251.6,114.7z\"/>\r\n<path class=\"st0\" d=\"M306.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19\r\n\tC298.1,95.6,306.6,104.2,306.6,114.7z\"/>\r\n<path d=\"M238,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4c1.3,0,2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4\r\n\tC238.1,99.8,238,99.8,238,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7\"/>\r\n<path d=\"M293,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4s2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4\r\n\tC293.1,99.8,293,99.8,293,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7\"/>\r\n<path class=\"st1\" d=\"M228.4,157.1h64.4c-15.4,63.8-25.1,105.4-25.1,151c0,8,0,69.5,23.4,69.5c12,0,22.2-10.8,22.2-20.5\r\n\tc0-2.8,0-4-4-12.5c-15.4-39.3-15.4-88.3-15.4-92.3c0-3.4,0-43.9,12-95.1h63.8c7.4,0,26.2,0,26.2-18.2c0-12.5-10.8-12.5-21.1-12.5\r\n\tH187.4c-13.1,0-32.5,0-58.7,27.9c-14.8,16.5-33,46.7-33,50.1s2.8,4.6,6.3,4.6c4,0,4.6-1.7,7.4-5.1c29.6-46.7,59.2-46.7,73.5-46.7\r\n\th32.5c-12.5,42.7-26.8,92.3-73.5,192c-4.6,9.1-4.6,10.3-4.6,13.7c0,12,10.3,14.8,15.4,14.8c16.5,0,21.1-14.8,27.9-38.7\r\n\tc9.1-29.1,9.1-30.2,14.8-53L228.4,157.1\"/>\r\n<path d=\"M246.3,149.1c5.5,0,1.9,0.1,13.5,0c10.3-0.1,12.4-0.1,15.7-0.2c0,0-0.3-3.1-0.3-3.1c-3.7,0.1-9.3,0.2-14.7,0.2\r\n\tc-3.9,0-8.5,0-14.6,0\"/>\r\n</svg>\r\n\r\nHere is my code that I get for my PsiCreatures_happy.svg also:\r\n\r\n<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\r\n<svg\r\n   xmlns:dc=\"http://purl.org/dc/elements/1.1/\"\r\n   xmlns:cc=\"http://creativecommons.org/ns#\"\r\n   xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\"\r\n   xmlns:svg=\"http://www.w3.org/2000/svg\"\r\n   xmlns=\"http://www.w3.org/2000/svg\"\r\n   xmlns:sodipodi=\"http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd\"\r\n   xmlns:inkscape=\"http://www.inkscape.org/namespaces/inkscape\"\r\n   width=\"210mm\"\r\n   height=\"297mm\"\r\n   viewBox=\"0 0 210 297\"\r\n   version=\"1.1\"\r\n   id=\"svg8\"\r\n   inkscape:version=\"1.0beta2 (2b71d25, 2019-12-03)\"\r\n   sodipodi:docname=\"psi_creature_happy.svg\">\r\n  <defs\r\n     id=\"defs2\" />\r\n  <sodipodi:namedview\r\n     id=\"base\"\r\n     pagecolor=\"#ffffff\"\r\n     bordercolor=\"#666666\"\r\n     borderopacity=\"1.0\"\r\n     inkscape:pageopacity=\"0.0\"\r\n     inkscape:pageshadow=\"2\"\r\n     inkscape:zoom=\"0.63261527\"\r\n     inkscape:cx=\"213.73446\"\r\n     inkscape:cy=\"398.71679\"\r\n     inkscape:document-units=\"mm\"\r\n     inkscape:current-layer=\"layer1\"\r\n     inkscape:document-rotation=\"0\"\r\n     showgrid=\"false\"\r\n     inkscape:window-width=\"1440\"\r\n     inkscape:window-height=\"805\"\r\n     inkscape:window-x=\"0\"\r\n     inkscape:window-y=\"23\"\r\n     inkscape:window-maximized=\"0\" />\r\n  <metadata\r\n     id=\"metadata5\">\r\n    <rdf:RDF>\r\n      <cc:Work\r\n         rdf:about=\"\">\r\n        <dc:format>image/svg+xml</dc:format>\r\n        <dc:type\r\n           rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\" />\r\n        <dc:title />\r\n      </cc:Work>\r\n    </rdf:RDF>\r\n  </metadata>\r\n  <g\r\n     inkscape:label=\"Layer 1\"\r\n     inkscape:groupmode=\"layer\"\r\n     id=\"layer1\">\r\n    <path\r\n       style=\"fill:#05d5fa;stroke:#000000;stroke-width:0.965;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1;fill-opacity:1;stroke-miterlimit:4;stroke-dasharray:none\"\r\n       d=\"m 86.543101,117.34948 -6.674715,-0.28525 -9.470109,-1.14097 -6.50357,-1.48327 -6.560616,-2.6813 -4.792104,-2.39605 -3.47998,-2.6813 -3.137686,-2.62424 -2.90949,-3.87933 L 41.189267,96.925991 38.507971,89.053248 37.823386,84.831636 37.1388,77.301187 36.796506,70.113032 36.62536,64.065856 36.454212,58.931457 35.65553,53.797062 34.457502,51.115766 32.746037,49.461351 31.20572,48.49152 28.296227,47.863982 h -0.627536 l 0.02889,-3.672229 2.253062,0.07815 3.765222,0.285244 2.795394,0.627536 2.567199,0.51344 3.594077,1.312124 2.738344,1.597367 2.681296,2.4531 2.396053,3.251785 2.224905,4.221614 0.855732,1.825565 1.140975,3.93637 0.798685,5.533736 0.45639,4.906201 0.228196,4.278664 0.171148,3.47998 0.285245,3.93637 0.741635,4.050469 0.741635,3.708175 1.312121,3.651126 1.54032,2.453101 1.711465,2.11081 1.93966,1.82556 2.05376,1.48327 2.90949,1.76852 2.738345,1.02688 3.137686,1.02687 3.993419,0.68459 3.02359,0.2282 3.93637,0.11409 0.114099,-46.437761 -0.399343,-3.87932 -0.399341,-3.308834 -1.198026,-2.567197 -2.167858,-2.224905 -2.96654,-1.711465 -3.308831,-0.684586 -3.308835,-0.399343 -3.87932,-0.285245 -1.198026,0.114099 -0.05705,-3.651126 55.679681,0.114097 0.17114,3.42293 -2.73834,0.114099 -3.19474,0.114096 -3.25178,0.570489 -3.19474,1.02688 -2.85244,1.768515 -1.65442,1.654416 -0.96983,1.825564 -0.68458,2.738345 -0.28525,3.194735 -0.22819,4.335711 -0.0571,5.248494 v 6.560617 l 0.0571,32.232605 h 3.99342 l 3.70817,-0.3423 3.99342,-0.74163 3.30883,-1.08393 3.53703,-1.65442 2.339,-1.65441 2.51015,-2.28196 2.4531,-2.96654 1.14098,-2.167856 1.59737,-4.050469 0.74163,-4.050466 0.57049,-5.077349 0.39934,-4.96325 0.3423,-6.674715 0.68458,-4.506857 0.62754,-4.164565 1.65442,-4.96325 1.25507,-2.852444 1.54032,-2.738345 1.99671,-2.624246 3.08064,-2.4531 3.13768,-1.711465 5.1344,-1.825564 3.47998,-0.399341 4.16457,-0.342294 1.59736,0.114099 -0.057,3.36588 -2.5672,0.51344 -2.51015,1.255075 -1.76852,2.053759 -0.79868,1.99671 -0.74164,3.137686 -0.28524,2.738345 -0.2282,3.194735 -0.057,2.624246 -0.57049,13.691724 -0.28525,3.651126 -0.62753,2.738345 -0.62754,4.107516 -1.65442,5.761934 -1.31212,2.795395 -1.71147,2.852441 -2.2249,2.62425 -2.5672,2.6813 -2.39605,1.76851 -3.25179,2.16786 -3.70817,1.93966 -4.56391,1.65441 -3.59408,1.08393 -3.82227,0.85573 -5.64783,1.02688 -5.41964,0.45639 -4.10752,0.39935 -1.82556,-0.17115 -0.17115,16.14482 0.34229,20.82283 0.3423,3.53703 0.51344,3.02359 0.85573,1.59737 1.48327,1.65441 1.93966,1.31213 1.71146,0.91278 1.99671,0.62753 2.51015,0.62754 3.30884,0.28525 3.99342,0.057 v 3.59407 l -55.736731,-0.11409 0.05705,-3.47998 4.735054,-0.17115 3.936371,-0.51344 3.365881,-1.08393 2.339004,-1.14098 1.312124,-0.96983 0.96983,-1.31212 0.96983,-1.48327 0.627536,-2.22491 0.57049,-3.99341 v -4.79211 l 0.171145,-14.88975 z\"\r\n       id=\"path24\"\r\n       inkscape:connector-curvature=\"0\" />\r\n    <path\r\n       style=\"fill:none;stroke:#000000;stroke-width:0.748;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1;stroke-miterlimit:4;stroke-dasharray:none\"\r\n       d=\"m 100.74351,45.639074 h 22.5961 c 0,0 -1.32918,-13.54538 -11.96264,-14.28422 -10.63346,-0.73884 -10.63346,14.28422 -10.63346,14.28422 z\"\r\n       id=\"path50\"\r\n       inkscape:connector-curvature=\"0\" />\r\n    <ellipse\r\n       style=\"fill:#050000;fill-opacity:1;fill-rule:evenodd;stroke-width:0.145492\"\r\n       id=\"path52\"\r\n       cx=\"111.24973\"\r\n       cy=\"39.960098\"\r\n       rx=\"3.1074717\"\r\n       ry=\"2.957778\" />\r\n    <path\r\n       inkscape:connector-curvature=\"0\"\r\n       id=\"path50-5\"\r\n       d=\"m 67.636488,45.513256 h 22.596103 c 0,0 -1.32918,-13.54538 -11.96264,-14.28422 -10.633463,-0.73884 -10.633463,14.28422 -10.633463,14.28422 z\"\r\n       style=\"fill:none;stroke:#000000;stroke-width:0.848;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1;stroke-miterlimit:4;stroke-dasharray:none\" />\r\n    <ellipse\r\n       ry=\"2.957778\"\r\n       rx=\"3.1074717\"\r\n       cy=\"39.477459\"\r\n       cx=\"78.607201\"\r\n       id=\"path52-8\"\r\n       style=\"fill:#050000;fill-opacity:1;fill-rule:evenodd;stroke-width:0.145492\" />\r\n    <path\r\n       style=\"fill:#050000;fill-opacity:1;stroke:#000000;stroke-width:0.108385px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1\"\r\n       d=\"m 75.959759,61.477064 c 13.457701,5.68591 26.622331,4.339439 39.675031,0.305194 -4.21382,3.690331 -9.14664,6.997189 -19.379726,7.477228 C 85.414893,68.929287 79.496059,65.864988 75.959759,61.477064 Z\"\r\n       id=\"path907\"\r\n       inkscape:connector-curvature=\"0\"\r\n       sodipodi:nodetypes=\"cccc\" />\r\n  </g>\r\n</svg>", "created_at": 1584599782, "updated_at": 1584599782}, "601013677": {"id": 601013677, "post_id": 583286873, "username": "luketimmerman314", "body": "Sorry, it looks all of the code wasn't posted. Let me try again. If it doesn't post this time, my question about building the file in Adobe Illustrator still stands.\r\n\r\n<?xml version=\"1.0\" encoding=\"utf-8\"?> \r\n<!-- Generator: Adobe Illustrator 21.0.2, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->\r\n<svg version=\"1.1\" id=\"Layer_1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" x=\"0px\" y=\"0px\"\r\n\t viewBox=\"0 0 480 480\" style=\"enable-background:new 0 0 480 480;\" xml:space=\"preserve\">\r\n<style type=\"text/css\">\r\n\t.st0{fill:#FFFFFF;}\r\n\t.st1{fill:#0C7F99;}\r\n</style>\r\n<path class=\"st0\" d=\"M251.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19\r\n\tC243.1,95.6,251.6,104.2,251.6,114.7z\"/>\r\n<path class=\"st0\" d=\"M306.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19\r\n\tC298.1,95.6,306.6,104.2,306.6,114.7z\"/>\r\n<path d=\"M238,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4c1.3,0,2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4\r\n\tC238.1,99.8,238,99.8,238,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7\"/>\r\n<path d=\"M293,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4s2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4\r\n\tC293.1,99.8,293,99.8,293,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7\"/>\r\n<path class=\"st1\" d=\"M228.4,157.1h64.4c-15.4,63.8-25.1,105.4-25.1,151c0,8,0,69.5,23.4,69.5c12,0,22.2-10.8,22.2-20.5\r\n\tc0-2.8,0-4-4-12.5c-15.4-39.3-15.4-88.3-15.4-92.3c0-3.4,0-43.9,12-95.1h63.8c7.4,0,26.2,0,26.2-18.2c0-12.5-10.8-12.5-21.1-12.5\r\n\tH187.4c-13.1,0-32.5,0-58.7,27.9c-14.8,16.5-33,46.7-33,50.1s2.8,4.6,6.3,4.6c4,0,4.6-1.7,7.4-5.1c29.6-46.7,59.2-46.7,73.5-46.7\r\n\th32.5c-12.5,42.7-26.8,92.3-73.5,192c-4.6,9.1-4.6,10.3-4.6,13.7c0,12,10.3,14.8,15.4,14.8c16.5,0,21.1-14.8,27.9-38.7\r\n\tc9.1-29.1,9.1-30.2,14.8-53L228.4,157.1\"/>\r\n<path d=\"M246.3,149.1c5.5,0,1.9,0.1,13.5,0c10.3-0.1,12.4-0.1,15.7-0.2c0,0-0.3-3.1-0.3-3.1c-3.7,0.1-9.3,0.2-14.7,0.2\r\n\tc-3.9,0-8.5,0-14.6,0\"/>\r\n</svg>", "created_at": 1584599927, "updated_at": 1584599927}, "601017925": {"id": 601017925, "post_id": 583286873, "username": "hanwenzhu", "body": "I'm not sure of that, but if PiCreatures_plain works on your `PsiCreature` class, I guess using Adobe Illustrator would make it work. If you want you can upload the SVG file to pastebin or GitHub gist.", "created_at": 1584600840, "updated_at": 1584600840}, "601297561": {"id": 601297561, "post_id": 583286873, "username": "luketimmerman314", "body": "Ok, thanks! I'll try it and see!", "created_at": 1584637040, "updated_at": 1584637040}, "583202077": {"id": 583202077, "post_id": 583202077, "username": "rpadams", "body": "Thanks for contributing to manim!\r\n\r\n**Please ensure that your pull request works with the latest version of manim.**\r\nYou should also include:\r\n\r\n1. The motivation for making this change (or link the relevant issues)\r\n\r\nI changed the color of the GraphScene axis but could not change the color of the tick labels to match.  This is a small change to graph_scene.py to provide the correct information to NumberLine.\r\n\r\n2. How you tested the new behavior (e.g. a minimal working example, before/after\r\nscreenshots, gifs, commands, etc.) This is rather informal at the moment, but\r\nthe goal is to show us how you know the pull request works as intended.\r\n\r\nBefore and after screenshots:\r\n<img width=\"628\" alt=\"before\" src=\"https://user-images.githubusercontent.com/1450174/76887507-c59c7f80-6858-11ea-9009-a98f6313bd6e.png\">\r\n<img width=\"697\" alt=\"after\" src=\"https://user-images.githubusercontent.com/1450174/76887510-c6cdac80-6858-11ea-8cdd-b183ef14dabd.png\">\r\n\r\n\r\n", "created_at": 1584468538, "updated_at": 1584468538}, "581833854": {"id": 581833854, "post_id": 581833854, "username": "daniel-lara-ec", "body": "This remove the decimal place by default and updates the label positions", "created_at": 1584307869, "updated_at": 1584307869}, "581411761": {"id": 581411761, "post_id": 581411761, "username": "emanueldias242", "body": "Microsoft Windows [vers\u00e3o 10.0.18362.657]\r\n(c) 2019 Microsoft Corporation. Todos os direitos reservados.\r\n\r\nC:\\Manim\\manim_3_feb>python -m manim exmaple_scenes.py SquareTocircle -pl\r\nTraceback (most recent call last):\r\n  File \"C:\\Users\\55359\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.7_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python37\\site-packages\\numpy\\core\\__init__.py\", line 16, in <module>\r\n    from . import multiarray\r\nImportError: DLL load failed: N\u00e3o foi poss\u00edvel encontrar o m\u00f3dulo especificado.\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.7_3.7.2032.0_x64__qbz5n2kfra8p0\\lib\\runpy.py\", line 193, in _run_module_as_main\r\n    \"__main__\", mod_spec)\r\n  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.7_3.7.2032.0_x64__qbz5n2kfra8p0\\lib\\runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"C:\\Manim\\manim_3_feb\\manim.py\", line 2, in <module>\r\n    import manimlib.config\r\n  File \"C:\\Manim\\manim_3_feb\\manimlib\\config.py\", line 8, in <module>\r\n    import manimlib.constants\r\n  File \"C:\\Manim\\manim_3_feb\\manimlib\\constants.py\", line 1, in <module>\r\n    import numpy as np\r\n  File \"C:\\Users\\55359\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.7_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python37\\site-packages\\numpy\\__init__.py\", line 142, in <module>\r\n    from . import add_newdocs\r\n  File \"C:\\Users\\55359\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.7_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python37\\site-packages\\numpy\\add_newdocs.py\", line 13, in <module>\r\n    from numpy.lib import add_newdoc\r\n  File \"C:\\Users\\55359\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.7_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python37\\site-packages\\numpy\\lib\\__init__.py\", line 8, in <module>\r\n    from .type_check import *\r\n  File \"C:\\Users\\55359\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.7_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python37\\site-packages\\numpy\\lib\\type_check.py\", line 11, in <module>\r\n    import numpy.core.numeric as _nx\r\n  File \"C:\\Users\\55359\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.7_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python37\\site-packages\\numpy\\core\\__init__.py\", line 26, in <module>\r\n    raise ImportError(msg)\r\nImportError:\r\nImporting the multiarray numpy extension module failed.  Most\r\nlikely you are trying to import a failed build of numpy.\r\nIf you're working with a numpy git repo, try `git clean -xdf` (removes all\r\nfiles not under version control).  Otherwise reinstall numpy.\r\n\r\nOriginal error was: DLL load failed: N\u00e3o foi poss\u00edvel encontrar o m\u00f3dulo especificado.\r\n![Screenshot_1](https://user-images.githubusercontent.com/59492262/76692039-b4bcf580-662f-11ea-8ec4-a3b40d370d44.png)\r\n\r\n\r\nI have no ideia what to do. Can someone help me? That happen after i try python -m manim exmaple_scenes.py SquareTocircle -pl  \r\n\r\nThanks", "created_at": 1584227402, "updated_at": 1584325665}, "599309498": {"id": 599309498, "post_id": 581411761, "username": "yalikes", "body": "according to the graph, I think you may need to reinstall numpy.\r\ntry run \r\n`pip uninstall numpy`\r\n`pip install numpy`", "created_at": 1584325665, "updated_at": 1584325665}, "580929459": {"id": 580929459, "post_id": 580929459, "username": "iGeek098", "body": "updated graph scale factor @ 10:29\r\n\r\njust perfect\r\n\r\nhad lots of fun\r\n\r\ncouldn't test yet!\r\n\r\nThanks for contributing to manim!\r\n\r\n**Please ensure that your pull request works with the latest version of manim.**\r\nYou should also include:\r\n\r\n1. The motivation for making this change (or link the relevant issues)\r\n2. How you tested the new behavior (e.g. a minimal working example, before/after\r\nscreenshots, gifs, commands, etc.) This is rather informal at the moment, but\r\nthe goal is to show us how you know the pull request works as intended.\r\n", "created_at": 1584142091, "updated_at": 1584142091}, "580052922": {"id": 580052922, "post_id": 580052922, "username": "luketimmerman314", "body": "Hello, I am using the blog Manim for Mac by bhowell4 to download the Manim library, and I am having trouble installing the requirements.txt program. I have no coding experience with this kind of thing so I am coming to y'all for help. I have attached screen shots of all of the code that my Terminal spit back out at me (it is a lot of screen shots, and I know that y'all don't need all of them, but I don't have the experience to know what y'all do and do not need). A little background: I successfully got through Step 6 of the walkthrough and am currently working in a virtual environment. I believe I successfully installed sox, FFmpeg, latex, and pycairo. I then got the error message when trying to install the requirements.txt program. To try and fix the problem, I tried upgrading pip, changing the version of opencv-python in the requirements.txt file from 3.4.2.17 to 4.2.0.32, and finally changed the Pillow from 5.2.0 in the same document to 6.2.0, but I ended up downloading version 7.0.0. Thanks in advance.\r\n\r\n![Screen Shot 2020-03-12 at 10 37 58 AM](https://user-images.githubusercontent.com/62107786/76542290-c8f7cb80-6452-11ea-9776-fe9f675a6f9c.png)\r\n![Screen Shot 2020-03-12 at 10 38 28 AM](https://user-images.githubusercontent.com/62107786/76542291-c9906200-6452-11ea-8c51-d9ad68c8a8d1.png)\r\n![Screen Shot 2020-03-12 at 10 38 53 AM](https://user-images.githubusercontent.com/62107786/76542292-c9906200-6452-11ea-9396-de403e5ac135.png)\r\n![Screen Shot 2020-03-12 at 10 39 51 AM](https://user-images.githubusercontent.com/62107786/76542294-ca28f880-6452-11ea-9e99-6d75a297d2d4.png)\r\n![Screen Shot 2020-03-12 at 10 40 22 AM](https://user-images.githubusercontent.com/62107786/76542297-ca28f880-6452-11ea-9bc1-42d0f2f11417.png)\r\n![Screen Shot 2020-03-12 at 10 40 47 AM](https://user-images.githubusercontent.com/62107786/76542298-cac18f00-6452-11ea-977e-c913c9751a2f.png)\r\n![Screen Shot 2020-03-12 at 10 41 12 AM](https://user-images.githubusercontent.com/62107786/76542299-cac18f00-6452-11ea-950e-7e8f9b35c8e1.png)\r\n![Screen Shot 2020-03-12 at 10 41 34 AM](https://user-images.githubusercontent.com/62107786/76542302-cac18f00-6452-11ea-81d0-383703d34f9b.png)\r\n![Screen Shot 2020-03-12 at 10 41 59 AM](https://user-images.githubusercontent.com/62107786/76542303-cb5a2580-6452-11ea-933b-60d1b45bc00f.png)\r\n![Screen Shot 2020-03-12 at 10 42 17 AM](https://user-images.githubusercontent.com/62107786/76542305-cb5a2580-6452-11ea-9eb6-a693150fa5ef.png)\r\n![Screen Shot 2020-03-12 at 10 42 32 AM](https://user-images.githubusercontent.com/62107786/76542308-cbf2bc00-6452-11ea-9558-3a8f180433eb.png)\r\n![Screen Shot 2020-03-12 at 10 42 48 AM](https://user-images.githubusercontent.com/62107786/76542311-cbf2bc00-6452-11ea-94f7-cbbffba40b29.png)\r\n![Screen Shot 2020-03-12 at 10 43 04 AM](https://user-images.githubusercontent.com/62107786/76542312-cbf2bc00-6452-11ea-807a-91cd4279a442.png)\r\n![Screen Shot 2020-03-12 at 10 43 20 AM](https://user-images.githubusercontent.com/62107786/76542316-cc8b5280-6452-11ea-8036-7e63bf10ac40.png)\r\n![Screen Shot 2020-03-12 at 10 43 47 AM](https://user-images.githubusercontent.com/62107786/76542317-cc8b5280-6452-11ea-912e-6b2a5687ba90.png)\r\n![Screen Shot 2020-03-12 at 10 44 11 AM](https://user-images.githubusercontent.com/62107786/76542319-cd23e900-6452-11ea-91fd-d90463823900.png)\r\n![Screen Shot 2020-03-12 at 10 44 27 AM](https://user-images.githubusercontent.com/62107786/76542321-cd23e900-6452-11ea-8441-7fd02e33efec.png)\r\n![Screen Shot 2020-03-12 at 10 44 44 AM](https://user-images.githubusercontent.com/62107786/76542324-cdbc7f80-6452-11ea-8983-b0625934a833.png)\r\n![Screen Shot 2020-03-12 at 10 45 02 AM](https://user-images.githubusercontent.com/62107786/76542325-ce551600-6452-11ea-9952-84cbe6afb369.png)\r\n![Screen Shot 2020-03-12 at 10 45 17 AM](https://user-images.githubusercontent.com/62107786/76542326-ce551600-6452-11ea-8722-14efbce4a6a7.png)\r\n![Screen Shot 2020-03-12 at 10 45 43 AM](https://user-images.githubusercontent.com/62107786/76542327-ceedac80-6452-11ea-9050-2930c76edcf7.png)\r\n![Screen Shot 2020-03-12 at 10 45 59 AM](https://user-images.githubusercontent.com/62107786/76542328-ceedac80-6452-11ea-821c-4fab826dcf14.png)\r\n![Screen Shot 2020-03-12 at 10 46 19 AM](https://user-images.githubusercontent.com/62107786/76542329-ceedac80-6452-11ea-8a44-7c5d0b1bfe8b.png)\r\n![Screen Shot 2020-03-12 at 10 46 38 AM](https://user-images.githubusercontent.com/62107786/76542330-cf864300-6452-11ea-96d7-66e2881fa995.png)\r\n![Screen Shot 2020-03-12 at 10 46 55 AM](https://user-images.githubusercontent.com/62107786/76542332-cf864300-6452-11ea-8267-c160fae937c7.png)\r\n![Screen Shot 2020-03-12 at 10 47 13 AM](https://user-images.githubusercontent.com/62107786/76542334-cf864300-6452-11ea-8ebe-51e358da371b.png)\r\n![Screen Shot 2020-03-12 at 10 47 31 AM](https://user-images.githubusercontent.com/62107786/76542336-d01ed980-6452-11ea-98ff-5ab26457351b.png)\r\n![Screen Shot 2020-03-12 at 10 47 50 AM](https://user-images.githubusercontent.com/62107786/76542338-d01ed980-6452-11ea-8768-6c844c94b343.png)\r\n![Screen Shot 2020-03-12 at 10 48 07 AM](https://user-images.githubusercontent.com/62107786/76542340-d0b77000-6452-11ea-8954-7747e1a1bf2a.png)\r\n![Screen Shot 2020-03-12 at 10 48 22 AM](https://user-images.githubusercontent.com/62107786/76542342-d0b77000-6452-11ea-969c-053bb305927b.png)\r\n![Screen Shot 2020-03-12 at 10 48 47 AM](https://user-images.githubusercontent.com/62107786/76542343-d1500680-6452-11ea-894e-69d2522a6623.png)\r\n![Screen Shot 2020-03-12 at 10 49 02 AM](https://user-images.githubusercontent.com/62107786/76542344-d1500680-6452-11ea-9dec-864c3c709ffe.png)\r\n![Screen Shot 2020-03-12 at 10 49 13 AM](https://user-images.githubusercontent.com/62107786/76542346-d1e89d00-6452-11ea-81a0-2aedc6d2621b.png)\r\n![Screen Shot 2020-03-12 at 10 49 35 AM](https://user-images.githubusercontent.com/62107786/76542347-d1e89d00-6452-11ea-8e63-cd5c9bab4e40.png)\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n", "created_at": 1584029801, "updated_at": 1584208615}, "599110339": {"id": 599110339, "post_id": 580052922, "username": "Aathish04", "body": "Manim isn't very picky when it comes to it's dependencies (at least as of now).\r\nIt might be worth trying to install Manim's dependencies without worrying about their versions.\r\nFirst, activate your Virtual Environment.\r\nThen, remove all previously installed packages, so we have a clean slate to work with.\r\n\r\n`pip3 uninstall -y $(pip3 freeze)`\r\n\r\nNow, copy and run this pip command, which basically just installs all of Manim's dependencies, without giving a damn about their versions.\r\n\r\n`pip3 install colour data decorator ffmpeg funcsigs future latex numpy opencv-python Pillow progressbar pycairo pydub scipy shutilwhich six sox tempdir tqdm`\r\n\r\nSee if all that works fine. It _should_, but if it doesn't, this might explain why:\r\n\r\n> **Update 02/19/20: This post is a little bit out of date with the current Manim library. For additional resources, please see https://repperiumsci.blogspot.com/2019/12/my-adventures-with-manim-part_22.html for additional ways to get Manim working on your computer.  Additionally, I have uploaded the version of Manim (09/18) pertinent to this this tutorial on GitHub. This should have been a fork, but it is what it is\u2026**\r\n-From his site\r\n\r\nSo maybe check out https://repperiumsci.blogspot.com/2019/12/my-adventures-with-manim-part_22.html and see if the instructions there work out.", "created_at": 1584208615, "updated_at": 1584208615}, "579660639": {"id": 579660639, "post_id": 579660639, "username": "ddxtanx", "body": "This is a merging of two pull requests to add better discontinuity detecting functionality and implicit function graphing capabilities in requests #832 and #839. I don't think it would serve much to repeat the rationale and testing again in another PR, so the relevant information is in the separate pull requests. If, however, it would be more conducive to reviewers to include such information here I would be happy to transfer it all over here. \r\n", "created_at": 1583982162, "updated_at": 1583982181}, "198259639": {"id": 198259639, "post_id": 198259639, "username": "Jspsun", "body": "Traceback (most recent call last):\r\n  File \"__main__.py\", line 17, in <module>\r\n    main()\r\n  File \"__main__.py\", line 12, in main\r\n    text = VoiceInput.getInput()\r\n  File \"/Users/Jspsun/AtomWorkspace/Athena/Source/VoiceInputHandler.py\", line 49, in getInput\r\n    return self.getText(audio)\r\n  File \"/Users/Jspsun/AtomWorkspace/Athena/Source/VoiceInputHandler.py\", line 38, in getText\r\n    text = self.r.recognize_google(audio)\r\n  File \"/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/speech_recognition/__init__.py\", line 738, in recognize_google\r\n    if len(actual_result.get(\"alternative\", [])) == 0: raise UnknownValueError()\r\nAttributeError: 'list' object has no attribute 'get'\r\n\r\nTo Reproduce:\r\n- Just make a horse sound. I'm not even kidding its the easiest way to reproduce the bug", "created_at": 1483256595, "updated_at": 1483256595}, "318514926": {"id": 318514926, "post_id": 318514926, "username": "Jli0423", "body": "Webpack dependency not installed with 'npm install'", "created_at": 1524855493, "updated_at": 1524855493}}